{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fc3fba43",
   "metadata": {},
   "source": [
    "# FUNGI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "35c82422",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "95e5bfde",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import packages\n",
    "import qiime2 as q2\n",
    "import os\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins import cutadapt\n",
    "from qiime2.plugins import dada2\n",
    "from qiime2.plugins import feature_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1fc05206",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: WORKDIR=/home/nezapa/qiime-thesis\n"
     ]
    }
   ],
   "source": [
    "# define workdir\n",
    "%env WORKDIR /home/nezapa/qiime-thesis\n",
    "WORKDIR = os.environ.get(\"WORKDIR\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed0ee88e",
   "metadata": {},
   "source": [
    "### IMPORT DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3b763efb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set manifest path\n",
    "manifest_path = './00.manifest.f.tsv'\n",
    "\n",
    "#import data\n",
    "single_end_demux = q2.Artifact.import_data('SampleData[SequencesWithQuality]', \n",
    "    view_type='SingleEndFastqManifestPhred33V2', \n",
    "    view=manifest_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b7dfe851",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-i2nlowie')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 97561e62-885d-4fb5-8e7c-6a3f0f120c18>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#summarise and visualise\n",
    "demux_summary = demux.visualizers.summarize(single_end_demux)\n",
    "demux_summary.visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "58f69d95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>location</th>\n",
       "      <th>origin</th>\n",
       "      <th>year</th>\n",
       "      <th>parkelj</th>\n",
       "      <th>specimen</th>\n",
       "      <th>population</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P01A</th>\n",
       "      <td>P01A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa200_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa200</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P03A</th>\n",
       "      <td>P03A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa201_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa201</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P05A</th>\n",
       "      <td>P05A</td>\n",
       "      <td>Stobe</td>\n",
       "      <td>Paa204_t10d</td>\n",
       "      <td>2016</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa204</td>\n",
       "      <td>dolenjska linija</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P07A</th>\n",
       "      <td>P07A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa209_t0</td>\n",
       "      <td>2017</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa209</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P09A</th>\n",
       "      <td>P09A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa210_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa210</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P11A</th>\n",
       "      <td>P11A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa211_t0</td>\n",
       "      <td>2018</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa211</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P13A</th>\n",
       "      <td>P13A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa219_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa219</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P15A</th>\n",
       "      <td>P15A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa220_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa220</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P17A</th>\n",
       "      <td>P17A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa221_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa221</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P19A</th>\n",
       "      <td>P19A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB271_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB271</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P20A</th>\n",
       "      <td>P20A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB272_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB272</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P21A</th>\n",
       "      <td>P21A</td>\n",
       "      <td>Jelsevnik</td>\n",
       "      <td>PB644_parkelj</td>\n",
       "      <td>2015</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB644</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P22A</th>\n",
       "      <td>P22A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PAP20_parkelj</td>\n",
       "      <td>2020</td>\n",
       "      <td>yes</td>\n",
       "      <td>PAP20</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample        location         origin  year parkelj specimen  \\\n",
       "sampleid                                                                \n",
       "P01A       P01A  Planinska_jama      Paa200_t0  2015      no   Paa200   \n",
       "P03A       P03A  Planinska_jama      Paa201_t0  2015      no   Paa201   \n",
       "P05A       P05A           Stobe    Paa204_t10d  2016      no   Paa204   \n",
       "P07A       P07A  Planinska_jama      Paa209_t0  2017      no   Paa209   \n",
       "P09A       P09A  Planinska_jama      Paa210_t0  2019      no   Paa210   \n",
       "P11A       P11A  Planinska_jama      Paa211_t0  2018      no   Paa211   \n",
       "P13A       P13A  Planinska_jama      Paa219_t0  2019      no   Paa219   \n",
       "P15A       P15A  Planinska_jama      Paa220_t0  2019      no   Paa220   \n",
       "P17A       P17A  Planinska_jama      Paa221_t0  2019      no   Paa221   \n",
       "P19A       P19A          Obrsec  PB271_parkelj  2016     yes    PB271   \n",
       "P20A       P20A          Obrsec  PB272_parkelj  2016     yes    PB272   \n",
       "P21A       P21A       Jelsevnik  PB644_parkelj  2015     yes    PB644   \n",
       "P22A       P22A          Obrsec  PAP20_parkelj  2020     yes    PAP20   \n",
       "\n",
       "                       population  \n",
       "sampleid                           \n",
       "P01A      linija reke Ljubljanice  \n",
       "P03A      linija reke Ljubljanice  \n",
       "P05A             dolenjska linija  \n",
       "P07A      linija reke Ljubljanice  \n",
       "P09A      linija reke Ljubljanice  \n",
       "P11A      linija reke Ljubljanice  \n",
       "P13A      linija reke Ljubljanice  \n",
       "P15A      linija reke Ljubljanice  \n",
       "P17A      linija reke Ljubljanice  \n",
       "P19A                훾rna podvrsta  \n",
       "P20A                훾rna podvrsta  \n",
       "P21A                훾rna podvrsta  \n",
       "P22A                훾rna podvrsta  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load and view metadata\n",
    "sample_metadata = q2.Metadata.load( f'{WORKDIR}/proteus_all/fungi/00.sample-metadata.tsv')\n",
    "sample_metadata.to_dataframe()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f16f02f",
   "metadata": {},
   "source": [
    "### TRIM PRIMERS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "09557e33",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P01A_0_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P01A_0_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P01A_0_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P01A_0_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.09 s (37 쨉s/read; 1.61 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  29,385\n",
      "Reads with adapters:                    29,266 (99.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        29,385 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,774,383 bp\n",
      "Total written (filtered):      8,015,046 bp (91.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 14 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 85.7%\n",
      "  C: 7.1%\n",
      "  G: 7.1%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t459.1\t0\t2\n",
      "5\t3\t28.7\t1\t1 2\n",
      "6\t1\t7.2\t1\t0 1\n",
      "25\t8\t0.0\t5\t8\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 29252 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t459.1\t0\t7\n",
      "4\t2\t114.8\t0\t0 2\n",
      "5\t2\t28.7\t1\t0 2\n",
      "6\t2\t7.2\t1\t0 2\n",
      "7\t1\t1.8\t1\t0 1\n",
      "19\t3\t0.0\t3\t1 1 0 1\n",
      "20\t2\t0.0\t4\t1 0 0 0 1\n",
      "21\t1\t0.0\t4\t1\n",
      "22\t8\t0.0\t4\t3 2 0 1 2\n",
      "23\t21\t0.0\t4\t3 3 5 10\n",
      "24\t83\t0.0\t4\t13 18 52\n",
      "25\t940\t0.0\t5\t195 735 8 1 1\n",
      "26\t28077\t0.0\t5\t27510 544 10 11 1 1\n",
      "27\t103\t0.0\t5\t27 73 2 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P03A_1_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P03A_1_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P03A_1_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P03A_1_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.03 s (25 쨉s/read; 2.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  41,937\n",
      "Reads with adapters:                    41,764 (99.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        41,937 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,576,495 bp\n",
      "Total written (filtered):     11,492,687 bp (91.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 33.3%\n",
      "  C: 66.7%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "4\t1\t163.8\t0\t1\n",
      "5\t2\t41.0\t1\t0 2\n",
      "7\t1\t2.6\t1\t0 1\n",
      "25\t1\t0.0\t5\t1\n",
      "59\t1\t0.0\t11\t0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 41758 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t655.3\t0\t14\n",
      "4\t2\t163.8\t0\t2\n",
      "5\t5\t41.0\t1\t1 4\n",
      "6\t2\t10.2\t1\t0 2\n",
      "8\t1\t0.6\t1\t0 0 1\n",
      "17\t1\t0.0\t3\t0 0 0 1\n",
      "18\t1\t0.0\t3\t1\n",
      "20\t6\t0.0\t4\t4 2\n",
      "21\t3\t0.0\t4\t1 1 1\n",
      "22\t2\t0.0\t4\t0 1 0 0 1\n",
      "23\t23\t0.0\t4\t4 7 2 10\n",
      "24\t110\t0.0\t4\t14 18 75 3\n",
      "25\t1321\t0.0\t5\t253 1046 21 1\n",
      "26\t40084\t0.0\t5\t39253 786 26 8 5 6\n",
      "27\t177\t0.0\t5\t44 128 4 1\n",
      "28\t2\t0.0\t5\t0 1 1\n",
      "29\t2\t0.0\t5\t0 0 0 2\n",
      "36\t1\t0.0\t7\t0 0 0 0 1\n",
      "39\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P05A_2_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P05A_2_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P05A_2_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P05A_2_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.08 s (38 쨉s/read; 1.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  28,470\n",
      "Reads with adapters:                    28,305 (99.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        28,470 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,505,043 bp\n",
      "Total written (filtered):      7,770,638 bp (91.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 15 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 46.7%\n",
      "  C: 40.0%\n",
      "  G: 13.3%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t444.8\t0\t6\n",
      "5\t4\t27.8\t1\t1 3\n",
      "7\t1\t1.7\t1\t0 1\n",
      "25\t4\t0.0\t5\t4\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 28290 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t444.8\t0\t4\n",
      "4\t2\t111.2\t0\t2\n",
      "5\t3\t27.8\t1\t0 3\n",
      "8\t6\t0.4\t1\t0 5 1\n",
      "18\t1\t0.0\t3\t1\n",
      "21\t1\t0.0\t4\t1\n",
      "22\t6\t0.0\t4\t0 3 0 1 2\n",
      "23\t21\t0.0\t4\t4 1 1 13 2\n",
      "24\t86\t0.0\t4\t13 14 59\n",
      "25\t857\t0.0\t5\t134 705 17 0 0 1\n",
      "26\t27158\t0.0\t5\t26580 560 9 4 4 1\n",
      "27\t142\t0.0\t5\t38 99 5\n",
      "29\t1\t0.0\t5\t0 0 0 1\n",
      "30\t1\t0.0\t6\t0 0 1\n",
      "33\t1\t0.0\t6\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P07A_3_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P07A_3_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P07A_3_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P07A_3_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.21 s (38 쨉s/read; 1.57 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  31,667\n",
      "Reads with adapters:                    31,509 (99.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        31,667 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,450,217 bp\n",
      "Total written (filtered):      8,632,612 bp (91.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 50.0%\n",
      "  C: 0.0%\n",
      "  G: 33.3%\n",
      "  T: 16.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t494.8\t0\t1\n",
      "4\t2\t123.7\t0\t2\n",
      "5\t1\t30.9\t1\t0 1\n",
      "25\t2\t0.0\t5\t2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 31503 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t8\t494.8\t0\t8\n",
      "5\t4\t30.9\t1\t1 3\n",
      "6\t4\t7.7\t1\t0 4\n",
      "8\t1\t0.5\t1\t0 1\n",
      "18\t3\t0.0\t3\t1 1 1\n",
      "20\t1\t0.0\t4\t0 0 0 1\n",
      "21\t2\t0.0\t4\t0 1 0 0 1\n",
      "22\t13\t0.0\t4\t4 2 1 2 4\n",
      "23\t18\t0.0\t4\t2 2 2 12\n",
      "24\t94\t0.0\t4\t19 18 55 2\n",
      "25\t1012\t0.0\t5\t172 827 12 0 1\n",
      "26\t30203\t0.0\t5\t29603 582 11 1 5 1\n",
      "27\t136\t0.0\t5\t36 98 1 0 0 1\n",
      "29\t1\t0.0\t5\t0 1\n",
      "34\t1\t0.0\t6\t1\n",
      "36\t1\t0.0\t7\t0 0 0 0 0 1\n",
      "42\t1\t0.0\t8\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P09A_4_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P09A_4_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P09A_4_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P09A_4_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.03 s (48 쨉s/read; 1.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  21,327\n",
      "Reads with adapters:                    21,179 (99.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        21,327 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,391,891 bp\n",
      "Total written (filtered):      5,842,523 bp (91.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 52.0%\n",
      "  C: 44.0%\n",
      "  G: 4.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t333.2\t0\t11\n",
      "5\t1\t20.8\t1\t0 1\n",
      "6\t1\t5.2\t1\t0 1\n",
      "25\t12\t0.0\t5\t11 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 21154 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t333.2\t0\t5\n",
      "5\t6\t20.8\t1\t2 4\n",
      "6\t1\t5.2\t1\t0 1\n",
      "17\t1\t0.0\t3\t0 0 1\n",
      "19\t1\t0.0\t3\t1\n",
      "22\t8\t0.0\t4\t2 1 1 0 4\n",
      "23\t13\t0.0\t4\t4 1 0 7 1\n",
      "24\t45\t0.0\t4\t6 8 31\n",
      "25\t645\t0.0\t5\t108 525 10 0 1 1\n",
      "26\t20326\t0.0\t5\t19914 399 8 4 1\n",
      "27\t103\t0.0\t5\t24 77 2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P11A_5_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P11A_5_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P11A_5_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P11A_5_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 0.87 s (89 쨉s/read; 0.67 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,720\n",
      "Reads with adapters:                     9,383 (96.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):         9,720 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,884,317 bp\n",
      "Total written (filtered):      2,641,228 bp (91.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 51 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.9%\n",
      "  C: 5.9%\n",
      "  G: 90.2%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t151.9\t0\t1\n",
      "5\t2\t9.5\t1\t0 2\n",
      "23\t1\t0.0\t4\t0 1\n",
      "24\t2\t0.0\t4\t0 2\n",
      "25\t45\t0.0\t5\t43 1 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 9332 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t151.9\t0\t11\n",
      "4\t3\t38.0\t0\t1 2\n",
      "5\t7\t9.5\t1\t0 7\n",
      "6\t2\t2.4\t1\t0 2\n",
      "9\t2\t0.0\t1\t0 0 2\n",
      "22\t2\t0.0\t4\t0 0 2\n",
      "23\t3\t0.0\t4\t1 0 2\n",
      "24\t21\t0.0\t4\t0 1 8 7 5\n",
      "25\t223\t0.0\t5\t0 35 86 101 1\n",
      "26\t9010\t0.0\t5\t1117 2416 5411 55 2 9\n",
      "27\t46\t0.0\t5\t4 8 13 21\n",
      "28\t1\t0.0\t5\t0 0 0 0 1\n",
      "48\t1\t0.0\t9\t0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P13A_6_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P13A_6_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P13A_6_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P13A_6_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.29 s (31 쨉s/read; 1.91 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  41,058\n",
      "Reads with adapters:                    40,909 (99.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        41,058 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,307,737 bp\n",
      "Total written (filtered):     11,246,026 bp (91.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 7 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.3%\n",
      "  C: 85.7%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t641.5\t0\t2\n",
      "5\t3\t40.1\t1\t0 3\n",
      "6\t1\t10.0\t1\t0 1\n",
      "25\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 40902 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t641.5\t0\t7\n",
      "4\t1\t160.4\t0\t0 1\n",
      "5\t6\t40.1\t1\t0 6\n",
      "8\t2\t0.6\t1\t0 0 2\n",
      "16\t1\t0.0\t3\t0 0 1\n",
      "18\t1\t0.0\t3\t1\n",
      "19\t1\t0.0\t3\t0 0 1\n",
      "21\t3\t0.0\t4\t2 1\n",
      "22\t9\t0.0\t4\t2 3 0 1 3\n",
      "23\t22\t0.0\t4\t3 4 1 14\n",
      "24\t117\t0.0\t4\t16 20 81\n",
      "25\t1308\t0.0\t5\t241 1051 14 1 0 1\n",
      "26\t39227\t0.0\t5\t38400 799 13 4 9 2\n",
      "27\t191\t0.0\t5\t51 124 2 0 0 14\n",
      "28\t2\t0.0\t5\t0 2\n",
      "29\t1\t0.0\t5\t0 1\n",
      "36\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "37\t1\t0.0\t7\t0 0 0 1\n",
      "43\t1\t0.0\t8\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P15A_7_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P15A_7_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P15A_7_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P15A_7_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.85 s (88 쨉s/read; 0.69 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,732\n",
      "Reads with adapters:                     9,586 (98.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):         9,732 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,907,856 bp\n",
      "Total written (filtered):      2,659,466 bp (91.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 84.0%\n",
      "  C: 16.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'A' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t152.1\t0\t1\n",
      "4\t8\t38.0\t0\t8\n",
      "5\t4\t9.5\t1\t0 4\n",
      "21\t1\t0.0\t4\t1\n",
      "23\t1\t0.0\t4\t1\n",
      "25\t9\t0.0\t5\t8 1\n",
      "80\t1\t0.0\t11\t0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 9561 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t152.1\t0\t2\n",
      "5\t5\t9.5\t1\t0 5\n",
      "6\t1\t2.4\t1\t0 1\n",
      "8\t2\t0.1\t1\t0 2\n",
      "20\t2\t0.0\t4\t1 0 0 1\n",
      "21\t2\t0.0\t4\t2\n",
      "22\t1\t0.0\t4\t0 0 0 1\n",
      "23\t8\t0.0\t4\t3 0 0 5\n",
      "24\t25\t0.0\t4\t1 1 22 1\n",
      "25\t339\t0.0\t5\t61 272 5 1\n",
      "26\t9128\t0.0\t5\t8824 283 14 5 2\n",
      "27\t46\t0.0\t5\t11 32 0 0 3\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P17A_8_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P17A_8_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P17A_8_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P17A_8_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 0.06 s (392 쨉s/read; 0.15 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     143\n",
      "Reads with adapters:                        63 (44.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):           143 (100.0%)\n",
      "\n",
      "Total basepairs processed:        40,153 bp\n",
      "Total written (filtered):         38,708 bp (96.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 50.0%\n",
      "  C: 16.7%\n",
      "  G: 16.7%\n",
      "  T: 16.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t2.2\t0\t1\n",
      "6\t3\t0.0\t1\t0 3\n",
      "25\t2\t0.0\t5\t2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 57 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "4\t1\t0.6\t0\t1\n",
      "5\t1\t0.1\t1\t1\n",
      "6\t3\t0.0\t1\t0 3\n",
      "25\t5\t0.0\t5\t1 4\n",
      "26\t47\t0.0\t5\t44 2 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P19A_9_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P19A_9_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P19A_9_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P19A_9_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.42 s (27 쨉s/read; 2.25 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  52,979\n",
      "Reads with adapters:                    52,778 (99.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        52,979 (100.0%)\n",
      "\n",
      "Total basepairs processed:    15,855,397 bp\n",
      "Total written (filtered):     14,485,601 bp (91.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 7 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 42.9%\n",
      "  C: 0.0%\n",
      "  G: 28.6%\n",
      "  T: 28.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t827.8\t0\t1\n",
      "5\t1\t51.7\t1\t0 1\n",
      "7\t2\t3.2\t1\t0 2\n",
      "25\t2\t0.0\t5\t2\n",
      "49\t1\t0.0\t9\t0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 52771 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9\t827.8\t0\t9\n",
      "4\t2\t206.9\t0\t0 2\n",
      "5\t2\t51.7\t1\t0 2\n",
      "6\t3\t12.9\t1\t0 3\n",
      "8\t2\t0.8\t1\t0 1 1\n",
      "18\t3\t0.0\t3\t2 0 1\n",
      "19\t2\t0.0\t3\t2\n",
      "20\t3\t0.0\t4\t2 1\n",
      "21\t4\t0.0\t4\t2 1 0 1\n",
      "22\t15\t0.0\t4\t3 4 1 3 4\n",
      "23\t31\t0.0\t4\t8 3 3 16 1\n",
      "24\t158\t0.0\t4\t19 23 114 2\n",
      "25\t1685\t0.0\t5\t315 1345 21 3 1\n",
      "26\t50638\t0.0\t5\t49644 962 15 8 4 5\n",
      "27\t209\t0.0\t5\t54 152 3\n",
      "28\t1\t0.0\t5\t1\n",
      "33\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "34\t1\t0.0\t6\t0 0 0 0 1\n",
      "36\t1\t0.0\t7\t0 0 0 0 0 1\n",
      "38\t1\t0.0\t7\t0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P20A_10_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P20A_10_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P20A_10_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P20A_10_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 0.08 s (208 쨉s/read; 0.29 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     365\n",
      "Reads with adapters:                       154 (42.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):           365 (100.0%)\n",
      "\n",
      "Total basepairs processed:       106,927 bp\n",
      "Total written (filtered):        103,506 bp (96.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 33.3%\n",
      "  C: 50.0%\n",
      "  G: 16.7%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t5.7\t0\t1\n",
      "4\t1\t1.4\t0\t1\n",
      "5\t1\t0.4\t1\t0 1\n",
      "6\t3\t0.1\t1\t0 3\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 148 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t5.7\t0\t6\n",
      "4\t5\t1.4\t0\t3 2\n",
      "5\t7\t0.4\t1\t1 6\n",
      "6\t2\t0.1\t1\t0 2\n",
      "8\t1\t0.0\t1\t0 1\n",
      "25\t4\t0.0\t5\t1 3\n",
      "26\t123\t0.0\t5\t119 4\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P21A_11_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P21A_11_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P21A_11_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P21A_11_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.53 s (27 쨉s/read; 2.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  55,883\n",
      "Reads with adapters:                    55,692 (99.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        55,883 (100.0%)\n",
      "\n",
      "Total basepairs processed:    16,737,077 bp\n",
      "Total written (filtered):     15,291,836 bp (91.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 18 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.2%\n",
      "  C: 22.2%\n",
      "  G: 33.3%\n",
      "  T: 22.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t873.2\t0\t4\n",
      "4\t4\t218.3\t0\t4\n",
      "5\t4\t54.6\t1\t0 4\n",
      "6\t1\t13.6\t1\t0 1\n",
      "7\t2\t3.4\t1\t0 2\n",
      "25\t3\t0.0\t5\t3\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 55674 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t873.2\t0\t4\n",
      "4\t2\t218.3\t0\t0 2\n",
      "5\t6\t54.6\t1\t0 6\n",
      "6\t3\t13.6\t1\t0 3\n",
      "7\t1\t3.4\t1\t0 1\n",
      "13\t1\t0.0\t2\t1\n",
      "18\t2\t0.0\t3\t1 0 0 1\n",
      "19\t1\t0.0\t3\t1\n",
      "20\t3\t0.0\t4\t2 1\n",
      "21\t8\t0.0\t4\t1 4 0 3\n",
      "22\t15\t0.0\t4\t3 3 0 1 8\n",
      "23\t27\t0.0\t4\t3 7 3 14\n",
      "24\t160\t0.0\t4\t16 27 115 2\n",
      "25\t1839\t0.0\t5\t327 1482 29 0 1\n",
      "26\t53328\t0.0\t5\t52191 1088 28 11 4 6\n",
      "27\t268\t0.0\t5\t85 181 2\n",
      "28\t2\t0.0\t5\t1 0 1\n",
      "29\t2\t0.0\t5\t0 0 0 2\n",
      "37\t1\t0.0\t7\t1\n",
      "46\t1\t0.0\t9\t0 0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P22A_12_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P22A_12_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-l6x5dsvc/P22A_12_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-wdeue40n/c619a7fd-9a40-4107-bd02-480e4e27ec83/data/P22A_12_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.05 s (48 쨉s/read; 1.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  21,820\n",
      "Reads with adapters:                    18,044 (82.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        21,820 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,323,910 bp\n",
      "Total written (filtered):      5,864,936 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 79 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 43.0%\n",
      "  C: 20.3%\n",
      "  G: 21.5%\n",
      "  T: 15.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t340.9\t0\t14\n",
      "4\t21\t85.2\t0\t21\n",
      "5\t14\t21.3\t1\t0 14\n",
      "6\t1\t5.3\t1\t0 1\n",
      "10\t2\t0.0\t2\t0 0 2\n",
      "11\t1\t0.0\t2\t0 0 1\n",
      "24\t1\t0.0\t4\t1\n",
      "25\t16\t0.0\t5\t13 0 0 3\n",
      "102\t9\t0.0\t11\t0 0 0 0 0 4 1 2 1 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 17965 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t134\t340.9\t0\t134\n",
      "4\t41\t85.2\t0\t22 19\n",
      "5\t112\t21.3\t1\t5 107\n",
      "6\t130\t5.3\t1\t7 123\n",
      "7\t8\t1.3\t1\t0 8\n",
      "8\t3\t0.3\t1\t0 0 3\n",
      "21\t2\t0.0\t4\t2\n",
      "22\t1\t0.0\t4\t1\n",
      "23\t10\t0.0\t4\t2 0 1 7\n",
      "24\t36\t0.0\t4\t7 3 26\n",
      "25\t538\t0.0\t5\t83 443 10 1 0 1\n",
      "26\t16868\t0.0\t5\t16338 507 13 6 1 3\n",
      "27\t80\t0.0\t5\t12 67 1\n",
      "31\t1\t0.0\t6\t1\n",
      "46\t1\t0.0\t9\t0 1\n"
     ]
    }
   ],
   "source": [
    "# adaoter has to be a reverse complement\n",
    "single_end_trimmed = cutadapt.methods.trim_single(\n",
    "    demultiplexed_sequences = single_end_demux,\n",
    "    front = ['TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT'],\n",
    "    adapter = ['AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC'], \n",
    "    error_rate = 0.2,\n",
    "    cores = 28\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9dc5fa1b",
   "metadata": {},
   "source": [
    "### START DADA2 MERGING AND CLUSTERING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5b9143d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada_single.R /tmp/qiime2-archive-buvx9ixb/785e3f81-e601-43c5-a033-2a7f2cb4a848/data /tmp/tmp9042hyth/output.tsv.biom /tmp/tmp9042hyth/track.tsv /tmp/tmp9042hyth 0 0 2.0 2 Inf pseudo consensus 1.0 14 1000000 NULL 16\n",
      "\n",
      "R version 4.1.3 (2022-03-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rcpp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DADA2: 1.22.0 / Rcpp: 1.0.8.3 / RcppParallel: 5.1.5 \n",
      "1) Filtering .............\n",
      "2) Learning Error Rates\n",
      "85087929 total bases in 311643 reads from 13 samples will be used for learning the error rates.\n",
      "3) Denoise samples .............\n",
      "  Pseudo-pool step .............\n",
      "4) Remove chimeras (method = consensus)\n",
      "5) Report read numbers through the pipeline\n",
      "6) Write output\n"
     ]
    }
   ],
   "source": [
    "# don't cut low quality ends!\n",
    "denoisetable_all, rep_seqs_all, denoising_stats_all = dada2.methods.denoise_single(\n",
    "    demultiplexed_seqs = single_end_trimmed.trimmed_sequences,\n",
    "    pooling_method = 'pseudo',\n",
    "    trunc_len = 0,\n",
    "    n_threads = 14\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ba41e22",
   "metadata": {},
   "source": [
    "### FILTERING BY ABUNDANCE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d4026bd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# based od Mock community results set the min_frequency\n",
    "denoisetable = feature_table.methods.filter_features(\n",
    "    table = denoisetable_all,\n",
    "    min_frequency = 20,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b91892ac",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-azpclkqw')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4bc423eb-6a4b-4429-b97d-4f2e734f1a79>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# summarise and visualise\n",
    "feature_table.visualizers.summarize(\n",
    "    table = denoisetable.filtered_table,\n",
    "    sample_metadata = sample_metadata\n",
    ").visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f4172b7e",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-hu2ce62p')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4e948ba8-5108-4b2e-a94e-c87b17467357>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# visualise the new set of representative sequences\n",
    "rep_seqs = feature_table.methods.filter_seqs(\n",
    "    data = rep_seqs_all,\n",
    "    table = denoisetable.filtered_table,\n",
    ")\n",
    "\n",
    "feature_table.visualizers.tabulate_seqs(data = rep_seqs.filtered_data).visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "df83dbec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'./results/denoising_stats.qza'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## save the output\n",
    "denoisetable.filtered_table.save('./results/denoisetable.qza')\n",
    "rep_seqs.filtered_data.save('./results/rep_seqs.qza')\n",
    "denoising_stats_all.save('./results/denoising_stats.qza')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
