{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "98c86edb",
   "metadata": {},
   "source": [
    "# ARCHAEA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "16aa43f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext autoreload\n",
    "%autoreload 2\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8c7aa148",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import packages\n",
    "import qiime2 as q2\n",
    "import os\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins import cutadapt\n",
    "from qiime2.plugins import dada2\n",
    "from qiime2.plugins import feature_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e63dc70b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: WORKDIR=/home/nezapa/qiime-thesis\n"
     ]
    }
   ],
   "source": [
    "# define workdir\n",
    "%env WORKDIR /home/nezapa/qiime-thesis\n",
    "WORKDIR = os.environ.get(\"WORKDIR\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6595d59d",
   "metadata": {},
   "source": [
    "### IMPORT DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e315ecec",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set manifest path\n",
    "manifest_path = './00.manifest.f.tsv'\n",
    "\n",
    "#import data\n",
    "single_end_demux = q2.Artifact.import_data('SampleData[SequencesWithQuality]', \n",
    "    view_type='SingleEndFastqManifestPhred33V2', \n",
    "    view=manifest_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4975e7ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-qz_s735n')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 0f686ea3-5c6e-41c6-a538-a7807f6bc427>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#summarise and visualise\n",
    "demux_summary = demux.visualizers.summarize(single_end_demux)\n",
    "demux_summary.visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "acc61315",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>location</th>\n",
       "      <th>origin</th>\n",
       "      <th>year</th>\n",
       "      <th>parkelj</th>\n",
       "      <th>specimen</th>\n",
       "      <th>population</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P01A</th>\n",
       "      <td>P01A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa200_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa200</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P03A</th>\n",
       "      <td>P03A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa201_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa201</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P05A</th>\n",
       "      <td>P05A</td>\n",
       "      <td>Stobe</td>\n",
       "      <td>Paa204_t10d</td>\n",
       "      <td>2016</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa204</td>\n",
       "      <td>dolenjska linija</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P07A</th>\n",
       "      <td>P07A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa209_t0</td>\n",
       "      <td>2017</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa209</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P09A</th>\n",
       "      <td>P09A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa210_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa210</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P11A</th>\n",
       "      <td>P11A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa211_t0</td>\n",
       "      <td>2018</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa211</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P13A</th>\n",
       "      <td>P13A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa219_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa219</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P15A</th>\n",
       "      <td>P15A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa220_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa220</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P17A</th>\n",
       "      <td>P17A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa221_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa221</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P19A</th>\n",
       "      <td>P19A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB271_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB271</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P20A</th>\n",
       "      <td>P20A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB272_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB272</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P21A</th>\n",
       "      <td>P21A</td>\n",
       "      <td>Jelsevnik</td>\n",
       "      <td>PB644_parkelj</td>\n",
       "      <td>2015</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB644</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P22A</th>\n",
       "      <td>P22A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PAP20_parkelj</td>\n",
       "      <td>2020</td>\n",
       "      <td>yes</td>\n",
       "      <td>PAP20</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample        location         origin  year parkelj specimen  \\\n",
       "sampleid                                                                \n",
       "P01A       P01A  Planinska_jama      Paa200_t0  2015      no   Paa200   \n",
       "P03A       P03A  Planinska_jama      Paa201_t0  2015      no   Paa201   \n",
       "P05A       P05A           Stobe    Paa204_t10d  2016      no   Paa204   \n",
       "P07A       P07A  Planinska_jama      Paa209_t0  2017      no   Paa209   \n",
       "P09A       P09A  Planinska_jama      Paa210_t0  2019      no   Paa210   \n",
       "P11A       P11A  Planinska_jama      Paa211_t0  2018      no   Paa211   \n",
       "P13A       P13A  Planinska_jama      Paa219_t0  2019      no   Paa219   \n",
       "P15A       P15A  Planinska_jama      Paa220_t0  2019      no   Paa220   \n",
       "P17A       P17A  Planinska_jama      Paa221_t0  2019      no   Paa221   \n",
       "P19A       P19A          Obrsec  PB271_parkelj  2016     yes    PB271   \n",
       "P20A       P20A          Obrsec  PB272_parkelj  2016     yes    PB272   \n",
       "P21A       P21A       Jelsevnik  PB644_parkelj  2015     yes    PB644   \n",
       "P22A       P22A          Obrsec  PAP20_parkelj  2020     yes    PAP20   \n",
       "\n",
       "                       population  \n",
       "sampleid                           \n",
       "P01A      linija reke Ljubljanice  \n",
       "P03A      linija reke Ljubljanice  \n",
       "P05A             dolenjska linija  \n",
       "P07A      linija reke Ljubljanice  \n",
       "P09A      linija reke Ljubljanice  \n",
       "P11A      linija reke Ljubljanice  \n",
       "P13A      linija reke Ljubljanice  \n",
       "P15A      linija reke Ljubljanice  \n",
       "P17A      linija reke Ljubljanice  \n",
       "P19A                훾rna podvrsta  \n",
       "P20A                훾rna podvrsta  \n",
       "P21A                훾rna podvrsta  \n",
       "P22A                훾rna podvrsta  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load and view metadata\n",
    "sample_metadata = q2.Metadata.load('../00.sample-metadata.tsv')\n",
    "sample_metadata.to_dataframe()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cdc2fab",
   "metadata": {},
   "source": [
    "### TRIM PRIMERS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5bfba5f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P01A_0_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P01A_0_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P01A_0_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P01A_0_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.50 s (26 쨉s/read; 2.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  57,816\n",
      "Reads with adapters:                     6,788 (11.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        57,816 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,741,485 bp\n",
      "Total written (filtered):     13,710,583 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 1410 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.0%\n",
      "  C: 34.1%\n",
      "  G: 27.9%\n",
      "  T: 11.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t511\t903.4\t0\t511\n",
      "4\t210\t225.8\t0\t101 109\n",
      "5\t390\t56.5\t1\t11 379\n",
      "6\t109\t14.1\t1\t4 105\n",
      "7\t76\t3.5\t1\t0 76\n",
      "8\t6\t0.9\t1\t0 5 1\n",
      "9\t16\t0.2\t1\t0 6 10\n",
      "10\t50\t0.1\t2\t0 43 7\n",
      "11\t33\t0.0\t2\t0 4 29\n",
      "12\t3\t0.0\t2\t0 0 3\n",
      "13\t2\t0.0\t2\t0 0 0 2\n",
      "14\t1\t0.0\t2\t0 0 0 1\n",
      "21\t3\t0.0\t4\t0 0 0 1 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 5378 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1886\t903.4\t0\t1886\n",
      "4\t601\t225.8\t0\t313 288\n",
      "5\t1365\t56.5\t1\t146 1219\n",
      "6\t1464\t14.1\t1\t43 1421\n",
      "8\t30\t0.9\t1\t0 30\n",
      "9\t32\t0.2\t1\t0 1 31\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P03A_1_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P03A_1_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P03A_1_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P03A_1_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.44 s (22 쨉s/read; 2.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  65,806\n",
      "Reads with adapters:                     7,982 (12.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        65,806 (100.0%)\n",
      "\n",
      "Total basepairs processed:    15,392,228 bp\n",
      "Total written (filtered):     15,358,097 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 1869 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.9%\n",
      "  C: 19.3%\n",
      "  G: 69.7%\n",
      "  T: 6.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1444\t1028.2\t0\t1444\n",
      "4\t101\t257.1\t0\t71 30\n",
      "5\t200\t64.3\t1\t7 193\n",
      "6\t80\t16.1\t1\t2 78\n",
      "7\t16\t4.0\t1\t0 16\n",
      "8\t2\t1.0\t1\t0 1 1\n",
      "9\t2\t0.3\t1\t0 0 2\n",
      "10\t6\t0.1\t2\t0 0 6\n",
      "11\t11\t0.0\t2\t0 0 11\n",
      "13\t1\t0.0\t2\t0 0 0 1\n",
      "20\t1\t0.0\t4\t0 0 0 0 1\n",
      "21\t5\t0.0\t4\t0 0 1 2 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 6113 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2188\t1028.2\t0\t2188\n",
      "4\t679\t257.1\t0\t321 358\n",
      "5\t1538\t64.3\t1\t130 1408\n",
      "6\t1629\t16.1\t1\t52 1577\n",
      "7\t1\t4.0\t1\t0 1\n",
      "8\t37\t1.0\t1\t0 37\n",
      "9\t39\t0.3\t1\t0 0 39\n",
      "26\t2\t0.0\t5\t2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P05A_2_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P05A_2_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P05A_2_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P05A_2_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.08 s (37 쨉s/read; 1.63 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  29,209\n",
      "Reads with adapters:                     3,641 (12.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        29,209 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,804,203 bp\n",
      "Total written (filtered):      7,787,669 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 982 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.5%\n",
      "  C: 29.7%\n",
      "  G: 12.2%\n",
      "  T: 39.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t143\t456.4\t0\t143\n",
      "4\t463\t114.1\t0\t348 115\n",
      "5\t164\t28.5\t1\t2 162\n",
      "6\t143\t7.1\t1\t5 138\n",
      "7\t26\t1.8\t1\t0 26\n",
      "8\t7\t0.4\t1\t0 5 2\n",
      "9\t5\t0.1\t1\t0 2 3\n",
      "10\t2\t0.0\t2\t0 0 2\n",
      "11\t28\t0.0\t2\t0 7 21\n",
      "21\t1\t0.0\t4\t0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 2659 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t933\t456.4\t0\t933\n",
      "4\t332\t114.1\t0\t174 158\n",
      "5\t685\t28.5\t1\t54 631\n",
      "6\t666\t7.1\t1\t9 657\n",
      "7\t1\t1.8\t1\t0 1\n",
      "8\t24\t0.4\t1\t0 21 3\n",
      "9\t16\t0.1\t1\t0 0 16\n",
      "26\t2\t0.0\t5\t2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P07A_3_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P07A_3_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P07A_3_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P07A_3_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.53 s (23 쨉s/read; 2.60 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  66,439\n",
      "Reads with adapters:                     7,606 (11.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        66,439 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,905,060 bp\n",
      "Total written (filtered):     14,869,153 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 1617 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 12.7%\n",
      "  C: 41.3%\n",
      "  G: 32.2%\n",
      "  T: 13.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t403\t1038.1\t0\t403\n",
      "4\t182\t259.5\t0\t124 58\n",
      "5\t500\t64.9\t1\t15 485\n",
      "6\t190\t16.2\t1\t3 187\n",
      "7\t106\t4.1\t1\t0 106\n",
      "8\t7\t1.0\t1\t0 7\n",
      "9\t20\t0.3\t1\t0 1 19\n",
      "10\t9\t0.1\t2\t0 4 5\n",
      "11\t197\t0.0\t2\t0 4 193\n",
      "13\t1\t0.0\t2\t0 0 1\n",
      "20\t1\t0.0\t4\t0 0 1\n",
      "21\t1\t0.0\t4\t1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 5989 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2086\t1038.1\t0\t2086\n",
      "4\t729\t259.5\t0\t371 358\n",
      "5\t1441\t64.9\t1\t149 1292\n",
      "6\t1644\t16.2\t1\t51 1593\n",
      "8\t53\t1.0\t1\t0 53\n",
      "9\t33\t0.3\t1\t0 0 33\n",
      "25\t1\t0.0\t5\t0 1\n",
      "26\t2\t0.0\t5\t2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P09A_4_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P09A_4_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P09A_4_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P09A_4_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.40 s (20 쨉s/read; 2.93 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  68,464\n",
      "Reads with adapters:                     7,341 (10.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        68,464 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,254,223 bp\n",
      "Total written (filtered):     14,220,940 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 945 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.9%\n",
      "  C: 28.1%\n",
      "  G: 36.8%\n",
      "  T: 18.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t259\t1069.8\t0\t259\n",
      "4\t123\t267.4\t0\t75 48\n",
      "5\t247\t66.9\t1\t6 241\n",
      "6\t87\t16.7\t1\t0 87\n",
      "7\t214\t4.2\t1\t0 214\n",
      "8\t3\t1.0\t1\t1 2\n",
      "9\t1\t0.3\t1\t0 0 1\n",
      "10\t6\t0.1\t2\t0 2 4\n",
      "11\t3\t0.0\t2\t0 0 3\n",
      "16\t1\t0.0\t3\t0 0 0 1\n",
      "21\t1\t0.0\t4\t0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 6396 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2272\t1069.8\t0\t2272\n",
      "4\t780\t267.4\t0\t370 410\n",
      "5\t1635\t66.9\t1\t163 1472\n",
      "6\t1617\t16.7\t1\t46 1571\n",
      "7\t1\t4.2\t1\t0 1\n",
      "8\t43\t1.0\t1\t0 43\n",
      "9\t48\t0.3\t1\t0 0 48\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P11A_5_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P11A_5_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P11A_5_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P11A_5_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.16 s (28 쨉s/read; 2.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  42,048\n",
      "Reads with adapters:                     4,280 (10.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        42,048 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,578,832 bp\n",
      "Total written (filtered):     10,559,512 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 259 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 24.3%\n",
      "  C: 39.0%\n",
      "  G: 18.5%\n",
      "  T: 18.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t60\t657.0\t0\t60\n",
      "4\t36\t164.2\t0\t21 15\n",
      "5\t105\t41.1\t1\t5 100\n",
      "6\t34\t10.3\t1\t0 34\n",
      "7\t13\t2.6\t1\t0 13\n",
      "8\t1\t0.6\t1\t0 1\n",
      "10\t6\t0.0\t2\t0 0 6\n",
      "11\t3\t0.0\t2\t0 0 3\n",
      "12\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 4021 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1412\t657.0\t0\t1412\n",
      "4\t483\t164.2\t0\t233 250\n",
      "5\t1038\t41.1\t1\t89 949\n",
      "6\t1028\t10.3\t1\t28 1000\n",
      "7\t3\t2.6\t1\t0 3\n",
      "8\t31\t0.6\t1\t0 31\n",
      "9\t24\t0.2\t1\t0 0 24\n",
      "26\t2\t0.0\t5\t1 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P13A_6_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P13A_6_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P13A_6_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P13A_6_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.26 s (29 쨉s/read; 2.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  43,096\n",
      "Reads with adapters:                     4,805 (11.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        43,096 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,949,523 bp\n",
      "Total written (filtered):     11,927,509 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 828 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 8.1%\n",
      "  C: 12.4%\n",
      "  G: 74.4%\n",
      "  T: 5.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t489\t673.4\t0\t489\n",
      "4\t85\t168.3\t0\t63 22\n",
      "5\t79\t42.1\t1\t2 77\n",
      "6\t37\t10.5\t1\t0 37\n",
      "7\t2\t2.6\t1\t0 2\n",
      "10\t1\t0.0\t2\t0 1\n",
      "11\t134\t0.0\t2\t0 0 134\n",
      "21\t1\t0.0\t4\t0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 3977 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1375\t673.4\t0\t1375\n",
      "4\t454\t168.3\t0\t224 230\n",
      "5\t1045\t42.1\t1\t93 952\n",
      "6\t1022\t10.5\t1\t33 989\n",
      "7\t1\t2.6\t1\t0 1\n",
      "8\t39\t0.7\t1\t0 38 1\n",
      "9\t36\t0.2\t1\t0 0 36\n",
      "26\t5\t0.0\t5\t5\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P15A_7_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P15A_7_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P15A_7_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P15A_7_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.19 s (23 쨉s/read; 2.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  52,400\n",
      "Reads with adapters:                     6,341 (12.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        52,400 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,201,655 bp\n",
      "Total written (filtered):     12,172,036 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 1515 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.5%\n",
      "  C: 54.5%\n",
      "  G: 11.7%\n",
      "  T: 17.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t232\t818.8\t0\t232\n",
      "4\t173\t204.7\t0\t119 54\n",
      "5\t382\t51.2\t1\t19 363\n",
      "6\t680\t12.8\t1\t4 676\n",
      "7\t14\t3.2\t1\t0 14\n",
      "8\t13\t0.8\t1\t0 13\n",
      "9\t3\t0.2\t1\t0 1 2\n",
      "10\t4\t0.0\t2\t0 2 2\n",
      "11\t9\t0.0\t2\t0 0 9\n",
      "12\t2\t0.0\t2\t0 0 2\n",
      "13\t2\t0.0\t2\t0 0 0 2\n",
      "21\t1\t0.0\t4\t0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 4826 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1675\t818.8\t0\t1675\n",
      "4\t547\t204.7\t0\t260 287\n",
      "5\t1264\t51.2\t1\t101 1163\n",
      "6\t1264\t12.8\t1\t44 1220\n",
      "8\t33\t0.8\t1\t0 33\n",
      "9\t41\t0.2\t1\t0 0 41\n",
      "26\t2\t0.0\t5\t2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P17A_8_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P17A_8_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P17A_8_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P17A_8_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.31 s (27 쨉s/read; 2.25 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  48,988\n",
      "Reads with adapters:                     5,661 (11.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        48,988 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,763,620 bp\n",
      "Total written (filtered):     10,739,072 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 1113 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 11.4%\n",
      "  C: 23.4%\n",
      "  G: 55.8%\n",
      "  T: 9.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t750\t765.4\t0\t750\n",
      "4\t107\t191.4\t0\t45 62\n",
      "5\t144\t47.8\t1\t15 129\n",
      "6\t54\t12.0\t1\t8 46\n",
      "7\t37\t3.0\t1\t0 37\n",
      "8\t5\t0.7\t1\t0 4 1\n",
      "9\t8\t0.2\t1\t0 1 7\n",
      "10\t1\t0.0\t2\t0 0 1\n",
      "11\t4\t0.0\t2\t0 3 1\n",
      "13\t2\t0.0\t2\t0 0 0 2\n",
      "15\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 4548 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1573\t765.4\t0\t1573\n",
      "4\t556\t191.4\t0\t265 291\n",
      "5\t1206\t47.8\t1\t113 1093\n",
      "6\t1168\t12.0\t1\t46 1122\n",
      "7\t1\t3.0\t1\t0 1\n",
      "8\t25\t0.7\t1\t0 24 1\n",
      "9\t18\t0.2\t1\t0 0 18\n",
      "10\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P19A_9_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P19A_9_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P19A_9_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P19A_9_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 0.99 s (34 쨉s/read; 1.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  29,582\n",
      "Reads with adapters:                     3,420 (11.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        29,582 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,532,045 bp\n",
      "Total written (filtered):      6,515,882 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 662 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 21.0%\n",
      "  C: 27.3%\n",
      "  G: 36.1%\n",
      "  T: 15.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t143\t462.2\t0\t143\n",
      "4\t107\t115.6\t0\t60 47\n",
      "5\t144\t28.9\t1\t9 135\n",
      "6\t67\t7.2\t1\t1 66\n",
      "7\t15\t1.8\t1\t0 15\n",
      "8\t169\t0.5\t1\t0 168 1\n",
      "9\t2\t0.1\t1\t0 0 2\n",
      "10\t4\t0.0\t2\t0 0 4\n",
      "11\t9\t0.0\t2\t0 0 9\n",
      "12\t2\t0.0\t2\t0 0 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 2758 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t914\t462.2\t0\t914\n",
      "4\t327\t115.6\t0\t155 172\n",
      "5\t717\t28.9\t1\t69 648\n",
      "6\t760\t7.2\t1\t17 743\n",
      "7\t3\t1.8\t1\t0 3\n",
      "8\t20\t0.5\t1\t0 20\n",
      "9\t16\t0.1\t1\t0 0 16\n",
      "26\t1\t0.0\t5\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P20A_10_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P20A_10_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P20A_10_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P20A_10_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.00 s (30 쨉s/read; 1.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  33,127\n",
      "Reads with adapters:                     3,603 (10.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        33,127 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,357,214 bp\n",
      "Total written (filtered):      7,340,828 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 576 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.8%\n",
      "  C: 22.7%\n",
      "  G: 25.9%\n",
      "  T: 18.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t123\t517.6\t0\t123\n",
      "4\t91\t129.4\t0\t43 48\n",
      "5\t229\t32.4\t1\t13 216\n",
      "6\t73\t8.1\t1\t3 70\n",
      "7\t39\t2.0\t1\t0 39\n",
      "8\t3\t0.5\t1\t0 3\n",
      "9\t6\t0.1\t1\t0 0 6\n",
      "10\t2\t0.0\t2\t0 0 2\n",
      "11\t10\t0.0\t2\t0 0 10\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 3027 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1062\t517.6\t0\t1062\n",
      "4\t350\t129.4\t0\t188 162\n",
      "5\t788\t32.4\t1\t84 704\n",
      "6\t793\t8.1\t1\t21 772\n",
      "8\t18\t0.5\t1\t0 18\n",
      "9\t15\t0.1\t1\t0 0 15\n",
      "26\t1\t0.0\t5\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P21A_11_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P21A_11_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P21A_11_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P21A_11_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.95 s (49 쨉s/read; 1.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  19,477\n",
      "Reads with adapters:                     2,301 (11.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        19,477 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,757,180 bp\n",
      "Total written (filtered):      4,746,630 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 518 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 31.1%\n",
      "  C: 14.3%\n",
      "  G: 22.4%\n",
      "  T: 32.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t85\t304.3\t0\t85\n",
      "4\t114\t76.1\t0\t40 74\n",
      "5\t196\t19.0\t1\t21 175\n",
      "6\t94\t4.8\t1\t5 89\n",
      "7\t14\t1.2\t1\t0 14\n",
      "8\t4\t0.3\t1\t0 4\n",
      "9\t3\t0.1\t1\t0 0 3\n",
      "10\t5\t0.0\t2\t0 0 5\n",
      "13\t1\t0.0\t2\t0 0 1\n",
      "17\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 1783 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t639\t304.3\t0\t639\n",
      "4\t187\t76.1\t0\t79 108\n",
      "5\t467\t19.0\t1\t50 417\n",
      "6\t465\t4.8\t1\t11 454\n",
      "8\t8\t0.3\t1\t0 7 1\n",
      "9\t15\t0.1\t1\t0 0 15\n",
      "26\t2\t0.0\t5\t2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P22A_12_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P22A_12_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-848o7xhu/P22A_12_L001_R1_001.fastq.gz --adapter AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT /tmp/qiime2-archive-6vcpveg9/1aa0f942-1bdc-49eb-a424-0aa286141f2f/data/P22A_12_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.08 s (31 쨉s/read; 1.91 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  34,294\n",
      "Reads with adapters:                     3,592 (10.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        34,294 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,078,224 bp\n",
      "Total written (filtered):      9,060,323 bp (99.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 59; Trimmed: 379 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10; 55-59 bp: 11\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.2%\n",
      "  C: 48.5%\n",
      "  G: 22.4%\n",
      "  T: 10.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t85\t535.8\t0\t85\n",
      "4\t167\t134.0\t0\t40 127\n",
      "5\t69\t33.5\t1\t0 69\n",
      "6\t37\t8.4\t1\t0 37\n",
      "7\t7\t2.1\t1\t0 7\n",
      "8\t2\t0.5\t1\t0 0 2\n",
      "9\t2\t0.1\t1\t0 1 1\n",
      "10\t2\t0.0\t2\t0 0 2\n",
      "11\t8\t0.0\t2\t0 0 8\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT; Type: regular 5'; Length: 59; Trimmed: 3213 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1126\t535.8\t0\t1126\n",
      "4\t351\t134.0\t0\t154 197\n",
      "5\t810\t33.5\t1\t75 735\n",
      "6\t799\t8.4\t1\t23 776\n",
      "8\t23\t0.5\t1\t0 22 1\n",
      "9\t17\t0.1\t1\t0 0 17\n",
      "24\t1\t0.0\t4\t0 0 1\n",
      "25\t3\t0.0\t5\t0 3\n",
      "26\t83\t0.0\t5\t80 3\n"
     ]
    }
   ],
   "source": [
    "# adaoter has to be a reverse complement\n",
    "single_end_trimmed = cutadapt.methods.trim_single(\n",
    "    demultiplexed_sequences = single_end_demux,\n",
    "    front = ['TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNAACTTTYRRCAAYGGATCWCT'],\n",
    "    adapter = ['AYTTAAGCATATCAATAAGCGGAGGCTGTCTCTTATACACATCTCCGAGCCCACGAGAC'], \n",
    "    error_rate = 0.2,\n",
    "    cores = 28\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c5ff28b",
   "metadata": {},
   "source": [
    "### START DADA2 MERGING AND CLUSTERING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7eb7935b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada_single.R /tmp/qiime2-archive-5wxzx__s/6573eabc-0d70-4f65-a86e-20dce1ff90bd/data /tmp/tmptt95w7j9/output.tsv.biom /tmp/tmptt95w7j9/track.tsv /tmp/tmptt95w7j9 260 0 2.0 2 Inf pseudo consensus 1.0 14 1000000 NULL 16\n",
      "\n",
      "R version 4.1.3 (2022-03-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rcpp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DADA2: 1.22.0 / Rcpp: 1.0.8.3 / RcppParallel: 5.1.5 \n",
      "1) Filtering .............\n",
      "2) Learning Error Rates\n",
      "69441580 total bases in 267083 reads from 13 samples will be used for learning the error rates.\n",
      "3) Denoise samples .............\n",
      "  Pseudo-pool step .............\n",
      "4) Remove chimeras (method = consensus)\n",
      "5) Report read numbers through the pipeline\n",
      "6) Write output\n"
     ]
    }
   ],
   "source": [
    "# set trunc-len based on the Interactive Quality Plot in datasummary.qzv\n",
    "# decide where the sequence quality drops below an acceptable level\n",
    "denoisetable_all, rep_seqs_all, denoising_stats_all = dada2.methods.denoise_single(\n",
    "    demultiplexed_seqs = single_end_trimmed.trimmed_sequences,\n",
    "    trunc_len = 260,\n",
    "    pooling_method = 'pseudo',\n",
    "    n_threads = 14\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c71ab7d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>filtered</th>\n",
       "      <th>percentage of input passed filter</th>\n",
       "      <th>denoised</th>\n",
       "      <th>non-chimeric</th>\n",
       "      <th>percentage of input non-chimeric</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample-id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P03A</th>\n",
       "      <td>65806.0</td>\n",
       "      <td>19633.0</td>\n",
       "      <td>29.83</td>\n",
       "      <td>11770.0</td>\n",
       "      <td>11761.0</td>\n",
       "      <td>17.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P20A</th>\n",
       "      <td>33127.0</td>\n",
       "      <td>9505.0</td>\n",
       "      <td>28.69</td>\n",
       "      <td>6162.0</td>\n",
       "      <td>6162.0</td>\n",
       "      <td>18.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P09A</th>\n",
       "      <td>68464.0</td>\n",
       "      <td>21983.0</td>\n",
       "      <td>32.11</td>\n",
       "      <td>15056.0</td>\n",
       "      <td>14747.0</td>\n",
       "      <td>21.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P17A</th>\n",
       "      <td>48988.0</td>\n",
       "      <td>15829.0</td>\n",
       "      <td>32.31</td>\n",
       "      <td>10985.0</td>\n",
       "      <td>10700.0</td>\n",
       "      <td>21.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P07A</th>\n",
       "      <td>66439.0</td>\n",
       "      <td>25750.0</td>\n",
       "      <td>38.76</td>\n",
       "      <td>15754.0</td>\n",
       "      <td>15718.0</td>\n",
       "      <td>23.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P19A</th>\n",
       "      <td>29582.0</td>\n",
       "      <td>11440.0</td>\n",
       "      <td>38.67</td>\n",
       "      <td>7174.0</td>\n",
       "      <td>7174.0</td>\n",
       "      <td>24.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P13A</th>\n",
       "      <td>43096.0</td>\n",
       "      <td>34143.0</td>\n",
       "      <td>79.23</td>\n",
       "      <td>11323.0</td>\n",
       "      <td>11003.0</td>\n",
       "      <td>25.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P21A</th>\n",
       "      <td>19477.0</td>\n",
       "      <td>9923.0</td>\n",
       "      <td>50.95</td>\n",
       "      <td>5943.0</td>\n",
       "      <td>5943.0</td>\n",
       "      <td>30.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P15A</th>\n",
       "      <td>52400.0</td>\n",
       "      <td>23700.0</td>\n",
       "      <td>45.23</td>\n",
       "      <td>16344.0</td>\n",
       "      <td>16241.0</td>\n",
       "      <td>30.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P22A</th>\n",
       "      <td>34294.0</td>\n",
       "      <td>22576.0</td>\n",
       "      <td>65.83</td>\n",
       "      <td>11096.0</td>\n",
       "      <td>11057.0</td>\n",
       "      <td>32.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P01A</th>\n",
       "      <td>57816.0</td>\n",
       "      <td>26405.0</td>\n",
       "      <td>45.67</td>\n",
       "      <td>18834.0</td>\n",
       "      <td>18820.0</td>\n",
       "      <td>32.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P11A</th>\n",
       "      <td>42048.0</td>\n",
       "      <td>25451.0</td>\n",
       "      <td>60.53</td>\n",
       "      <td>13995.0</td>\n",
       "      <td>13823.0</td>\n",
       "      <td>32.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P05A</th>\n",
       "      <td>29209.0</td>\n",
       "      <td>20745.0</td>\n",
       "      <td>71.02</td>\n",
       "      <td>15153.0</td>\n",
       "      <td>15118.0</td>\n",
       "      <td>51.76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             input  filtered  percentage of input passed filter  denoised  \\\n",
       "sample-id                                                                   \n",
       "P03A       65806.0   19633.0                              29.83   11770.0   \n",
       "P20A       33127.0    9505.0                              28.69    6162.0   \n",
       "P09A       68464.0   21983.0                              32.11   15056.0   \n",
       "P17A       48988.0   15829.0                              32.31   10985.0   \n",
       "P07A       66439.0   25750.0                              38.76   15754.0   \n",
       "P19A       29582.0   11440.0                              38.67    7174.0   \n",
       "P13A       43096.0   34143.0                              79.23   11323.0   \n",
       "P21A       19477.0    9923.0                              50.95    5943.0   \n",
       "P15A       52400.0   23700.0                              45.23   16344.0   \n",
       "P22A       34294.0   22576.0                              65.83   11096.0   \n",
       "P01A       57816.0   26405.0                              45.67   18834.0   \n",
       "P11A       42048.0   25451.0                              60.53   13995.0   \n",
       "P05A       29209.0   20745.0                              71.02   15153.0   \n",
       "\n",
       "           non-chimeric  percentage of input non-chimeric  \n",
       "sample-id                                                  \n",
       "P03A            11761.0                             17.87  \n",
       "P20A             6162.0                             18.60  \n",
       "P09A            14747.0                             21.54  \n",
       "P17A            10700.0                             21.84  \n",
       "P07A            15718.0                             23.66  \n",
       "P19A             7174.0                             24.25  \n",
       "P13A            11003.0                             25.53  \n",
       "P21A             5943.0                             30.51  \n",
       "P15A            16241.0                             30.99  \n",
       "P22A            11057.0                             32.24  \n",
       "P01A            18820.0                             32.55  \n",
       "P11A            13823.0                             32.87  \n",
       "P05A            15118.0                             51.76  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# view denoising stats\n",
    "stats_df = denoising_stats_all.view(q2.Metadata).to_dataframe()\n",
    "#stats_df\n",
    "#stats_df.sort_values('percentage of input passed filter')\n",
    "stats_df.sort_values('percentage of input non-chimeric')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6171842b",
   "metadata": {},
   "source": [
    "### FILTERING BY ABUNDANCE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ce0dfb09",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set the min_frequency\n",
    "denoisetable= feature_table.methods.filter_features(\n",
    "    table = denoisetable_all,\n",
    "    min_frequency = 20,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c0629cf3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-w6jpk4q8')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 7d3c7702-5015-4e43-9086-e9b3b61f1fff>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# summarise and visualise\n",
    "feature_table.visualizers.summarize(\n",
    "    table = denoisetable.filtered_table,\n",
    "    sample_metadata = sample_metadata\n",
    ").visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "73415312",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-zbudpye7')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: ed090428-7efd-451a-a3ec-0ba8e3079105>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# visualise the new set of representative sequences\n",
    "rep_seqs = feature_table.methods.filter_seqs(\n",
    "    data = rep_seqs_all,\n",
    "    table = denoisetable.filtered_table,\n",
    ")\n",
    "\n",
    "feature_table.visualizers.tabulate_seqs(data = rep_seqs.filtered_data).visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e8d2220c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'./results/denoising_stats.qza'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## save the output\n",
    "denoisetable.filtered_table.save('./results/denoisetable.qza')\n",
    "rep_seqs.filtered_data.save('./results/rep_seqs.qza')\n",
    "denoising_stats_all.save('./results/denoising_stats.qza')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
