{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9b636f2c",
   "metadata": {},
   "source": [
    "# EUKARYA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d12f2127",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "%reload_ext autoreload\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "09ddae8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import packages\n",
    "import qiime2 as q2\n",
    "import os\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins import cutadapt\n",
    "from qiime2.plugins import dada2\n",
    "from qiime2.plugins import feature_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "89b3a580",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: WORKDIR=/home/nezapa/qiime-thesis\n"
     ]
    }
   ],
   "source": [
    "# define workdir\n",
    "%env WORKDIR /home/nezapa/qiime-thesis\n",
    "WORKDIR = os.environ.get(\"WORKDIR\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8302d782",
   "metadata": {},
   "source": [
    "### IMPORT DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c7b36a4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set manifest path\n",
    "manifest_path = './00.manifest.f.tsv'\n",
    "\n",
    "#import data\n",
    "single_end_demux = q2.Artifact.import_data('SampleData[SequencesWithQuality]', \n",
    "    view_type='SingleEndFastqManifestPhred33V2', \n",
    "    view=manifest_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "44ef14ee",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-4kump978')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 8788f93e-ca28-423b-bdaf-2324f15f7589>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#summarise and visualise\n",
    "demux_summary = demux.visualizers.summarize(single_end_demux)\n",
    "demux_summary.visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "23e2d0f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>location</th>\n",
       "      <th>origin</th>\n",
       "      <th>year</th>\n",
       "      <th>parkelj</th>\n",
       "      <th>specimen</th>\n",
       "      <th>population</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P01A</th>\n",
       "      <td>P01A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa200_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa200</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P03A</th>\n",
       "      <td>P03A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa201_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa201</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P05A</th>\n",
       "      <td>P05A</td>\n",
       "      <td>Stobe</td>\n",
       "      <td>Paa204_t10d</td>\n",
       "      <td>2016</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa204</td>\n",
       "      <td>dolenjska linija</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P07A</th>\n",
       "      <td>P07A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa209_t0</td>\n",
       "      <td>2017</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa209</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P09A</th>\n",
       "      <td>P09A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa210_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa210</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P11A</th>\n",
       "      <td>P11A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa211_t0</td>\n",
       "      <td>2018</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa211</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P13A</th>\n",
       "      <td>P13A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa219_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa219</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P15A</th>\n",
       "      <td>P15A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa220_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa220</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P17A</th>\n",
       "      <td>P17A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa221_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa221</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P19A</th>\n",
       "      <td>P19A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB271_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB271</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P20A</th>\n",
       "      <td>P20A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB272_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB272</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P21A</th>\n",
       "      <td>P21A</td>\n",
       "      <td>Jelsevnik</td>\n",
       "      <td>PB644_parkelj</td>\n",
       "      <td>2015</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB644</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P22A</th>\n",
       "      <td>P22A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PAP20_parkelj</td>\n",
       "      <td>2020</td>\n",
       "      <td>yes</td>\n",
       "      <td>PAP20</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample        location         origin  year parkelj specimen  \\\n",
       "sampleid                                                                \n",
       "P01A       P01A  Planinska_jama      Paa200_t0  2015      no   Paa200   \n",
       "P03A       P03A  Planinska_jama      Paa201_t0  2015      no   Paa201   \n",
       "P05A       P05A           Stobe    Paa204_t10d  2016      no   Paa204   \n",
       "P07A       P07A  Planinska_jama      Paa209_t0  2017      no   Paa209   \n",
       "P09A       P09A  Planinska_jama      Paa210_t0  2019      no   Paa210   \n",
       "P11A       P11A  Planinska_jama      Paa211_t0  2018      no   Paa211   \n",
       "P13A       P13A  Planinska_jama      Paa219_t0  2019      no   Paa219   \n",
       "P15A       P15A  Planinska_jama      Paa220_t0  2019      no   Paa220   \n",
       "P17A       P17A  Planinska_jama      Paa221_t0  2019      no   Paa221   \n",
       "P19A       P19A          Obrsec  PB271_parkelj  2016     yes    PB271   \n",
       "P20A       P20A          Obrsec  PB272_parkelj  2016     yes    PB272   \n",
       "P21A       P21A       Jelsevnik  PB644_parkelj  2015     yes    PB644   \n",
       "P22A       P22A          Obrsec  PAP20_parkelj  2020     yes    PAP20   \n",
       "\n",
       "                       population  \n",
       "sampleid                           \n",
       "P01A      linija reke Ljubljanice  \n",
       "P03A      linija reke Ljubljanice  \n",
       "P05A             dolenjska linija  \n",
       "P07A      linija reke Ljubljanice  \n",
       "P09A      linija reke Ljubljanice  \n",
       "P11A      linija reke Ljubljanice  \n",
       "P13A      linija reke Ljubljanice  \n",
       "P15A      linija reke Ljubljanice  \n",
       "P17A      linija reke Ljubljanice  \n",
       "P19A                훾rna podvrsta  \n",
       "P20A                훾rna podvrsta  \n",
       "P21A                훾rna podvrsta  \n",
       "P22A                훾rna podvrsta  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load and view metadata\n",
    "sample_metadata = q2.Metadata.load('../00.sample-metadata.tsv')\n",
    "sample_metadata.to_dataframe()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a49bd1d0",
   "metadata": {},
   "source": [
    "### TRIM PRIMERS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7676902c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P01A_0_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P01A_0_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P01A_0_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P01A_0_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.23 s (25 쨉s/read; 2.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  49,871\n",
      "Reads with adapters:                    48,485 (97.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        49,871 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,952,993 bp\n",
      "Total written (filtered):     13,841,076 bp (92.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 155 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.1%\n",
      "  C: 20.0%\n",
      "  G: 50.3%\n",
      "  T: 11.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t8\t779.2\t0\t8\n",
      "4\t4\t194.8\t0\t1 3\n",
      "5\t68\t48.7\t1\t3 65\n",
      "6\t16\t12.2\t1\t2 14\n",
      "7\t2\t3.0\t1\t0 2\n",
      "8\t6\t0.8\t1\t0 6\n",
      "9\t7\t0.2\t1\t0 1 6\n",
      "10\t6\t0.0\t2\t0 1 5\n",
      "11\t4\t0.0\t2\t0 1 3\n",
      "16\t1\t0.0\t3\t0 0 0 1\n",
      "17\t4\t0.0\t3\t4\n",
      "60\t1\t0.0\t10\t0 0 1\n",
      "296\t11\t0.0\t10\t0 3 6 0 1 0 0 0 0 1\n",
      "297\t16\t0.0\t10\t0 3 4 6 1 0 0 0 1 0 1\n",
      "298\t1\t0.0\t10\t0 0 0 0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 48330 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t133\t779.2\t0\t133\n",
      "4\t48\t194.8\t0\t38 10\n",
      "5\t70\t48.7\t1\t4 66\n",
      "6\t8\t12.2\t1\t0 8\n",
      "7\t3\t3.0\t1\t0 3\n",
      "15\t1\t0.0\t3\t1\n",
      "16\t1\t0.0\t3\t0 0 0 1\n",
      "18\t2\t0.0\t3\t0 0 1 1\n",
      "19\t7\t0.0\t3\t0 0 2 2 3\n",
      "20\t42\t0.0\t4\t4 2 23 13\n",
      "21\t319\t0.0\t4\t25 110 176 7 1\n",
      "22\t3107\t0.0\t4\t751 2316 40\n",
      "23\t44492\t0.0\t4\t42306 2143 26 10 7\n",
      "24\t94\t0.0\t4\t21 71 2\n",
      "25\t1\t0.0\t5\t0 0 0 0 1\n",
      "27\t1\t0.0\t5\t1\n",
      "35\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P03A_1_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P03A_1_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P03A_1_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P03A_1_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.19 s (38 쨉s/read; 1.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  31,396\n",
      "Reads with adapters:                    31,072 (99.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        31,396 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,411,159 bp\n",
      "Total written (filtered):      8,699,972 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 67 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 23.9%\n",
      "  C: 37.3%\n",
      "  G: 22.4%\n",
      "  T: 16.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t490.6\t0\t13\n",
      "4\t12\t122.6\t0\t6 6\n",
      "5\t16\t30.7\t1\t2 14\n",
      "6\t14\t7.7\t1\t0 14\n",
      "7\t2\t1.9\t1\t0 2\n",
      "10\t2\t0.0\t2\t0 1 1\n",
      "11\t1\t0.0\t2\t0 0 1\n",
      "18\t1\t0.0\t3\t1\n",
      "78\t2\t0.0\t10\t0 2\n",
      "79\t1\t0.0\t10\t0 0 0 0 0 0 1\n",
      "297\t3\t0.0\t10\t0 1 1 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 31005 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t490.6\t0\t19\n",
      "4\t9\t122.6\t0\t6 3\n",
      "5\t15\t30.7\t1\t1 14\n",
      "6\t2\t7.7\t1\t0 2\n",
      "16\t1\t0.0\t3\t0 1\n",
      "17\t2\t0.0\t3\t1 0 0 1\n",
      "18\t1\t0.0\t3\t0 1\n",
      "19\t2\t0.0\t3\t1 0 0 1\n",
      "20\t31\t0.0\t4\t2 6 11 12\n",
      "21\t213\t0.0\t4\t15 73 123 2\n",
      "22\t2036\t0.0\t4\t463 1542 30 1\n",
      "23\t28622\t0.0\t4\t26914 1674 23 9 2\n",
      "24\t50\t0.0\t4\t11 39\n",
      "35\t2\t0.0\t7\t0 0 0 0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P05A_2_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P05A_2_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P05A_2_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P05A_2_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.02 s (44 쨉s/read; 1.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  23,073\n",
      "Reads with adapters:                    22,938 (99.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        23,073 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,920,813 bp\n",
      "Total written (filtered):      6,395,445 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 13 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.4%\n",
      "  C: 15.4%\n",
      "  G: 53.8%\n",
      "  T: 15.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t360.5\t0\t4\n",
      "5\t4\t22.5\t1\t1 3\n",
      "6\t1\t5.6\t1\t0 1\n",
      "8\t3\t0.4\t1\t0 3\n",
      "296\t1\t0.0\t10\t0 0 0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 22925 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t8\t360.5\t0\t8\n",
      "4\t5\t90.1\t0\t4 1\n",
      "5\t8\t22.5\t1\t0 8\n",
      "6\t2\t5.6\t1\t0 2\n",
      "16\t1\t0.0\t3\t0 1\n",
      "18\t1\t0.0\t3\t0 0 0 1\n",
      "19\t4\t0.0\t3\t0 1 2 1\n",
      "20\t22\t0.0\t4\t0 2 5 15\n",
      "21\t144\t0.0\t4\t13 55 74 2\n",
      "22\t1502\t0.0\t4\t334 1136 31 1\n",
      "23\t21176\t0.0\t4\t20006 1136 23 5 6\n",
      "24\t52\t0.0\t4\t8 38 6\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P07A_3_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P07A_3_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P07A_3_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P07A_3_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.00 s (47 쨉s/read; 1.27 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  21,077\n",
      "Reads with adapters:                    21,017 (99.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        21,077 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,307,867 bp\n",
      "Total written (filtered):      5,823,298 bp (92.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 45 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 66.7%\n",
      "  C: 11.1%\n",
      "  G: 15.6%\n",
      "  T: 6.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t329.3\t0\t1\n",
      "5\t5\t20.6\t1\t0 5\n",
      "6\t2\t5.1\t1\t0 2\n",
      "9\t1\t0.1\t1\t0 0 1\n",
      "17\t3\t0.0\t3\t3\n",
      "76\t11\t0.0\t10\t0 9 2\n",
      "77\t2\t0.0\t10\t0 0 2\n",
      "78\t10\t0.0\t10\t0 9 1\n",
      "79\t2\t0.0\t10\t0 1 0 0 0 0 1\n",
      "80\t1\t0.0\t10\t0 0 1\n",
      "114\t1\t0.0\t10\t0 0 0 0 1\n",
      "295\t2\t0.0\t10\t0 0 1 1\n",
      "296\t2\t0.0\t10\t0 0 0 0 0 0 0 0 0 2\n",
      "297\t2\t0.0\t10\t0 1 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 20972 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t329.3\t0\t3\n",
      "4\t2\t82.3\t0\t1 1\n",
      "5\t1\t20.6\t1\t0 1\n",
      "17\t2\t0.0\t3\t1 0 0 0 1\n",
      "18\t2\t0.0\t3\t0 1 0 1\n",
      "19\t2\t0.0\t3\t1 1\n",
      "20\t22\t0.0\t4\t1 3 10 8\n",
      "21\t137\t0.0\t4\t11 46 80\n",
      "22\t1365\t0.0\t4\t322 1021 22\n",
      "23\t19399\t0.0\t4\t18370 1009 16 2 2\n",
      "24\t34\t0.0\t4\t8 23 3\n",
      "31\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "35\t2\t0.0\t7\t0 0 0 0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P09A_4_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P09A_4_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P09A_4_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P09A_4_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.00 s (49 쨉s/read; 1.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  20,350\n",
      "Reads with adapters:                    20,296 (99.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        20,350 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,103,676 bp\n",
      "Total written (filtered):      5,637,859 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 19 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 57.9%\n",
      "  C: 15.8%\n",
      "  G: 26.3%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t318.0\t0\t2\n",
      "4\t1\t79.5\t0\t0 1\n",
      "5\t5\t19.9\t1\t3 2\n",
      "6\t1\t5.0\t1\t0 1\n",
      "7\t1\t1.2\t1\t0 1\n",
      "9\t1\t0.1\t1\t0 0 1\n",
      "17\t3\t0.0\t3\t3\n",
      "34\t1\t0.0\t6\t0 1\n",
      "296\t1\t0.0\t10\t0 1\n",
      "297\t3\t0.0\t10\t0 0 1 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 20277 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t318.0\t0\t6\n",
      "4\t2\t79.5\t0\t2\n",
      "5\t5\t19.9\t1\t1 4\n",
      "7\t1\t1.2\t1\t0 1\n",
      "17\t1\t0.0\t3\t0 0 0 1\n",
      "18\t1\t0.0\t3\t1\n",
      "19\t6\t0.0\t3\t1 1 2 2\n",
      "20\t15\t0.0\t4\t1 4 6 3 1\n",
      "21\t147\t0.0\t4\t17 48 80 2\n",
      "22\t1296\t0.0\t4\t306 964 25 1\n",
      "23\t18768\t0.0\t4\t17680 1065 17 3 3\n",
      "24\t27\t0.0\t4\t2 23 2\n",
      "34\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "36\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P11A_5_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P11A_5_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P11A_5_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P11A_5_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.05 s (38 쨉s/read; 1.59 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  27,808\n",
      "Reads with adapters:                    27,761 (99.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        27,808 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,340,786 bp\n",
      "Total written (filtered):      7,703,631 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 12 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.0%\n",
      "  C: 0.0%\n",
      "  G: 66.7%\n",
      "  T: 8.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t434.5\t0\t2\n",
      "4\t2\t108.6\t0\t2\n",
      "5\t2\t27.2\t1\t0 2\n",
      "8\t1\t0.4\t1\t0 1\n",
      "34\t1\t0.0\t6\t0 1\n",
      "297\t4\t0.0\t10\t0 1 1 1 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 27749 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t434.5\t0\t3\n",
      "5\t3\t27.2\t1\t1 2\n",
      "6\t1\t6.8\t1\t0 1\n",
      "18\t1\t0.0\t3\t0 0 1\n",
      "19\t8\t0.0\t3\t2 2 1 1 2\n",
      "20\t13\t0.0\t4\t0 0 6 7\n",
      "21\t220\t0.0\t4\t20 65 126 4 5\n",
      "22\t1734\t0.0\t4\t401 1206 50 75 2\n",
      "23\t25722\t0.0\t4\t22596 1474 1555 89 8\n",
      "24\t43\t0.0\t4\t8 30 3 2\n",
      "35\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P13A_6_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P13A_6_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P13A_6_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P13A_6_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 0.99 s (35 쨉s/read; 1.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  28,469\n",
      "Reads with adapters:                    28,418 (99.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        28,469 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,533,237 bp\n",
      "Total written (filtered):      7,879,628 bp (92.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 27 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 66.7%\n",
      "  C: 7.4%\n",
      "  G: 11.1%\n",
      "  T: 14.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1\t27.8\t1\t0 1\n",
      "17\t2\t0.0\t3\t2\n",
      "39\t1\t0.0\t7\t0 0 1\n",
      "40\t9\t0.0\t8\t0 6 2 1\n",
      "41\t1\t0.0\t8\t0 1\n",
      "54\t1\t0.0\t10\t0 0 0 1\n",
      "76\t2\t0.0\t10\t0 2\n",
      "77\t1\t0.0\t10\t0 1\n",
      "78\t2\t0.0\t10\t0 2\n",
      "295\t1\t0.0\t10\t0 0 0 0 0 0 0 1\n",
      "296\t3\t0.0\t10\t0 0 1 1 0 0 0 0 0 1\n",
      "297\t3\t0.0\t10\t0 2 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 28391 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t444.8\t0\t6\n",
      "4\t3\t111.2\t0\t2 1\n",
      "5\t1\t27.8\t1\t0 1\n",
      "6\t1\t7.0\t1\t0 1\n",
      "13\t1\t0.0\t2\t1\n",
      "16\t1\t0.0\t3\t0 0 1\n",
      "19\t3\t0.0\t3\t0 2 0 1\n",
      "20\t21\t0.0\t4\t1 3 9 7 1\n",
      "21\t177\t0.0\t4\t17 60 96 4\n",
      "22\t1781\t0.0\t4\t411 1326 39 2 3\n",
      "23\t26346\t0.0\t4\t24793 1510 37 4 2\n",
      "24\t47\t0.0\t4\t6 38 3\n",
      "25\t2\t0.0\t5\t0 0 0 2\n",
      "35\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P15A_7_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P15A_7_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P15A_7_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P15A_7_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.99 s (63 쨉s/read; 0.95 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  15,637\n",
      "Reads with adapters:                    15,599 (99.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        15,637 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,686,763 bp\n",
      "Total written (filtered):      4,329,031 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 11 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 45.5%\n",
      "  C: 9.1%\n",
      "  G: 36.4%\n",
      "  T: 9.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t244.3\t0\t1\n",
      "5\t3\t15.3\t1\t0 3\n",
      "6\t1\t3.8\t1\t0 1\n",
      "7\t1\t1.0\t1\t0 1\n",
      "9\t1\t0.1\t1\t0 0 1\n",
      "33\t1\t0.0\t6\t0 0 0 1\n",
      "78\t1\t0.0\t10\t0 0 1\n",
      "79\t1\t0.0\t10\t0 0 0 1\n",
      "297\t1\t0.0\t10\t0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 15588 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t244.3\t0\t5\n",
      "4\t1\t61.1\t0\t0 1\n",
      "5\t3\t15.3\t1\t0 3\n",
      "19\t3\t0.0\t3\t0 1 1 1\n",
      "20\t9\t0.0\t4\t0 2 2 5\n",
      "21\t78\t0.0\t4\t7 24 47\n",
      "22\t970\t0.0\t4\t230 726 13 0 1\n",
      "23\t14500\t0.0\t4\t13726 759 12 1 2\n",
      "24\t19\t0.0\t4\t5 13 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P17A_8_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P17A_8_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P17A_8_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P17A_8_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 0.98 s (46 쨉s/read; 1.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  21,407\n",
      "Reads with adapters:                    21,364 (99.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        21,407 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,418,786 bp\n",
      "Total written (filtered):      5,928,745 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 17 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 52.9%\n",
      "  C: 35.3%\n",
      "  G: 5.9%\n",
      "  T: 5.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t6\t20.9\t1\t1 5\n",
      "7\t1\t1.3\t1\t0 1\n",
      "9\t1\t0.1\t1\t0 1\n",
      "17\t3\t0.0\t3\t3\n",
      "76\t1\t0.0\t10\t0 1\n",
      "78\t1\t0.0\t10\t0 1\n",
      "79\t2\t0.0\t10\t0 0 2\n",
      "81\t1\t0.0\t10\t0 1\n",
      "296\t1\t0.0\t10\t0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 21347 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t334.5\t0\t2\n",
      "5\t1\t20.9\t1\t0 1\n",
      "16\t1\t0.0\t3\t0 0 0 1\n",
      "18\t2\t0.0\t3\t0 0 1 1\n",
      "19\t3\t0.0\t3\t0 1 0 2\n",
      "20\t23\t0.0\t4\t0 2 12 9\n",
      "21\t130\t0.0\t4\t10 43 75 1 1\n",
      "22\t1346\t0.0\t4\t306 1019 17 4\n",
      "23\t19803\t0.0\t4\t18732 1054 15 1 1\n",
      "24\t36\t0.0\t4\t6 29 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P19A_9_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P19A_9_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P19A_9_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P19A_9_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.01 s (42 쨉s/read; 1.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  23,977\n",
      "Reads with adapters:                    23,939 (99.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        23,977 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,192,385 bp\n",
      "Total written (filtered):      6,642,609 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 10 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 10.0%\n",
      "  C: 20.0%\n",
      "  G: 50.0%\n",
      "  T: 20.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t3\t23.4\t1\t0 3\n",
      "6\t1\t5.9\t1\t0 1\n",
      "61\t1\t0.0\t10\t0 0 1\n",
      "64\t1\t0.0\t10\t0 0 1\n",
      "296\t1\t0.0\t10\t0 1\n",
      "297\t3\t0.0\t10\t0 2 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 23929 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1\t23.4\t1\t0 1\n",
      "16\t1\t0.0\t3\t0 0 0 1\n",
      "18\t2\t0.0\t3\t0 1 1\n",
      "19\t4\t0.0\t3\t1 2 1\n",
      "20\t23\t0.0\t4\t4 4 6 9\n",
      "21\t160\t0.0\t4\t15 65 79 1\n",
      "22\t1545\t0.0\t4\t347 1163 34 1\n",
      "23\t22143\t0.0\t4\t20617 1499 21 5 1\n",
      "24\t49\t0.0\t4\t8 40 1\n",
      "35\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P20A_10_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P20A_10_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P20A_10_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P20A_10_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.09 s (39 쨉s/read; 1.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  27,674\n",
      "Reads with adapters:                    27,602 (99.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        27,674 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,296,551 bp\n",
      "Total written (filtered):      7,656,658 bp (92.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 50 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.0%\n",
      "  C: 26.0%\n",
      "  G: 36.0%\n",
      "  T: 24.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t432.4\t0\t1\n",
      "4\t1\t108.1\t0\t0 1\n",
      "5\t15\t27.0\t1\t2 13\n",
      "6\t1\t6.8\t1\t0 1\n",
      "17\t1\t0.0\t3\t1\n",
      "18\t2\t0.0\t3\t0 0 0 2\n",
      "288\t1\t0.0\t10\t0 0 0 0 0 0 1\n",
      "294\t1\t0.0\t10\t0 0 0 1\n",
      "295\t1\t0.0\t10\t0 0 1\n",
      "296\t9\t0.0\t10\t0 7 1 0 0 0 0 0 0 0 1\n",
      "297\t17\t0.0\t10\t0 8 5 2 0 0 0 0 0 0 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 27552 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t432.4\t0\t6\n",
      "4\t1\t108.1\t0\t1\n",
      "5\t6\t27.0\t1\t1 5\n",
      "6\t1\t6.8\t1\t0 1\n",
      "16\t1\t0.0\t3\t0 0 1\n",
      "17\t1\t0.0\t3\t0 0 1\n",
      "18\t2\t0.0\t3\t2\n",
      "19\t2\t0.0\t3\t0 0 1 1\n",
      "20\t32\t0.0\t4\t3 6 14 9\n",
      "21\t177\t0.0\t4\t18 60 90 8 1\n",
      "22\t1847\t0.0\t4\t427 1390 29 1\n",
      "23\t25418\t0.0\t4\t23939 1442 21 14 2\n",
      "24\t58\t0.0\t4\t8 48 2\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P21A_11_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P21A_11_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P21A_11_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P21A_11_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.00 s (43 쨉s/read; 1.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  23,275\n",
      "Reads with adapters:                    23,203 (99.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        23,275 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,963,220 bp\n",
      "Total written (filtered):      6,423,535 bp (92.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 72 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 43.1%\n",
      "  C: 13.9%\n",
      "  G: 29.2%\n",
      "  T: 13.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t363.7\t0\t4\n",
      "5\t4\t22.7\t1\t0 4\n",
      "6\t2\t5.7\t1\t0 2\n",
      "8\t1\t0.4\t1\t0 1\n",
      "17\t1\t0.0\t3\t1\n",
      "43\t2\t0.0\t8\t0 0 0 0 1 1\n",
      "54\t1\t0.0\t10\t0 1\n",
      "75\t2\t0.0\t10\t0 2\n",
      "76\t13\t0.0\t10\t0 12 1\n",
      "77\t1\t0.0\t10\t0 1\n",
      "78\t7\t0.0\t10\t0 6 0 1\n",
      "79\t5\t0.0\t10\t0 2 2 0 1\n",
      "81\t3\t0.0\t10\t0 0 2 1\n",
      "82\t1\t0.0\t10\t0 0 1\n",
      "120\t1\t0.0\t10\t0 0 0 0 1\n",
      "158\t1\t0.0\t10\t0 1\n",
      "206\t1\t0.0\t10\t0 0 0 1\n",
      "295\t3\t0.0\t10\t0 0 0 1 1 0 0 0 0 1\n",
      "296\t6\t0.0\t10\t0 1 3 2\n",
      "297\t11\t0.0\t10\t0 8 0 1 0 0 0 0 0 1 1\n",
      "298\t2\t0.0\t10\t0 1 0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 23131 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t363.7\t0\t5\n",
      "4\t5\t90.9\t0\t4 1\n",
      "5\t4\t22.7\t1\t0 4\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t1\t0.0\t3\t0 0 0 1\n",
      "18\t3\t0.0\t3\t1 0 0 2\n",
      "19\t3\t0.0\t3\t0 0 1 2\n",
      "20\t18\t0.0\t4\t0 4 5 9\n",
      "21\t142\t0.0\t4\t10 54 76 2\n",
      "22\t1435\t0.0\t4\t331 1069 32 2 1\n",
      "23\t21467\t0.0\t4\t20209 1232 18 6 2\n",
      "24\t46\t0.0\t4\t8 36 2\n",
      "35\t1\t0.0\t7\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P22A_12_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P22A_12_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-pnnd6eea/P22A_12_L001_R1_001.fastq.gz --adapter AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG /tmp/qiime2-archive-ybsx1o92/6f1979f4-bae3-46a5-8fef-b3b8824e2ffc/data/P22A_12_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.06 s (27 쨉s/read; 2.25 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  39,904\n",
      "Reads with adapters:                    39,793 (99.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        39,904 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,928,563 bp\n",
      "Total written (filtered):     11,012,097 bp (92.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 51; Trimmed: 79 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 55.7%\n",
      "  C: 15.2%\n",
      "  G: 12.7%\n",
      "  T: 16.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t623.5\t0\t5\n",
      "4\t6\t155.9\t0\t1 5\n",
      "5\t6\t39.0\t1\t2 4\n",
      "6\t1\t9.7\t1\t0 1\n",
      "7\t2\t2.4\t1\t0 2\n",
      "9\t2\t0.2\t1\t0 0 2\n",
      "17\t5\t0.0\t3\t5\n",
      "23\t22\t0.0\t4\t22\n",
      "26\t2\t0.0\t5\t2\n",
      "41\t1\t0.0\t8\t0 0 1\n",
      "65\t1\t0.0\t10\t0 0 1\n",
      "78\t1\t0.0\t10\t0 0 0 1\n",
      "88\t1\t0.0\t10\t0 0 0 0 1\n",
      "98\t1\t0.0\t10\t0 0 0 0 1\n",
      "101\t1\t0.0\t10\t0 0 0 1\n",
      "102\t3\t0.0\t10\t0 0 0 3\n",
      "151\t2\t0.0\t10\t0 0 0 0 1 0 1\n",
      "174\t2\t0.0\t10\t0 0 0 0 0 0 2\n",
      "175\t2\t0.0\t10\t0 0 0 0 0 0 2\n",
      "243\t1\t0.0\t10\t0 0 0 1\n",
      "295\t1\t0.0\t10\t0 0 0 1\n",
      "296\t5\t0.0\t10\t0 0 4 0 1\n",
      "297\t6\t0.0\t10\t0 3 2 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG; Type: regular 5'; Length: 56; Trimmed: 39714 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-51 bp: 10\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t623.5\t0\t11\n",
      "4\t2\t155.9\t0\t2\n",
      "5\t4\t39.0\t1\t1 3\n",
      "13\t1\t0.0\t2\t1\n",
      "18\t2\t0.0\t3\t0 0 2\n",
      "19\t3\t0.0\t3\t0 0 1 2\n",
      "20\t26\t0.0\t4\t1 4 9 8 4\n",
      "21\t254\t0.0\t4\t10 75 127 41 1\n",
      "22\t2410\t0.0\t4\t344 1358 695 11 2\n",
      "23\t36927\t0.0\t4\t20222 16127 543 31 4\n",
      "24\t73\t0.0\t4\t5 34 34\n",
      "25\t1\t0.0\t5\t0 0 1\n"
     ]
    }
   ],
   "source": [
    "# adapter has to be a reverse complement\n",
    "single_end_trimmed = cutadapt.methods.trim_single(\n",
    "    demultiplexed_sequences = single_end_demux,\n",
    "    front = ['TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNTTAAARVGYTCGTAGTYG'],\n",
    "    adapter = ['AYTTRAAGDAATTGACGGTGTCTCTTATACACATCTCCGAGCCCACGAGAC'], \n",
    "    error_rate = 0.2,\n",
    "    cores = 28\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a396f1ba",
   "metadata": {},
   "source": [
    "### START DADA2 MERGING AND CLUSTERING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0a0cce12",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada_single.R /tmp/qiime2-archive-9t3loal3/790c3619-98b1-466e-9458-6b23912ab5b2/data /tmp/tmps347ax26/output.tsv.biom /tmp/tmps347ax26/track.tsv /tmp/tmps347ax26 260 0 2.0 2 Inf pseudo consensus 1.0 14 1000000 NULL 16\n",
      "\n",
      "R version 4.1.3 (2022-03-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rcpp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DADA2: 1.22.0 / Rcpp: 1.0.8.3 / RcppParallel: 5.1.5 \n",
      "1) Filtering .............\n",
      "2) Learning Error Rates\n",
      "86697000 total bases in 333450 reads from 13 samples will be used for learning the error rates.\n",
      "3) Denoise samples .............\n",
      "  Pseudo-pool step .............\n",
      "4) Remove chimeras (method = consensus)\n",
      "5) Report read numbers through the pipeline\n",
      "6) Write output\n"
     ]
    }
   ],
   "source": [
    "# set trunc-len based on the Interactive Quality Plot in datasummary.qzv\n",
    "# decide where the sequence quality drops below an acceptable level\n",
    "denoisetable_all, rep_seqs_all, denoising_stats_all = dada2.methods.denoise_single(\n",
    "    demultiplexed_seqs = single_end_trimmed.trimmed_sequences,\n",
    "    trunc_len = 260,\n",
    "    pooling_method = 'pseudo',\n",
    "    n_threads = 14\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0314801d",
   "metadata": {},
   "source": [
    "### FILTERING BY ABUNDANCE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a5539775",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set the min_frequency\n",
    "denoisetable= feature_table.methods.filter_features(\n",
    "    table = denoisetable_all,\n",
    "    min_frequency = 20,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "1d588c37",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-52xfe2u7')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: ab5d1481-e0e2-411c-9218-d54a2d837be6>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# summarise and visualise\n",
    "q2.plugins.feature_table.visualizers.summarize(\n",
    "    table = denoisetable.filtered_table,\n",
    "    sample_metadata = sample_metadata\n",
    ").visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f20ab134",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-k3xk5pc9')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: ee410509-36c3-48f1-9838-7c9de03bb676>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# visualise the new set of representative sequences\n",
    "rep_seqs = feature_table.methods.filter_seqs(\n",
    "    data = rep_seqs_all,\n",
    "    table = denoisetable.filtered_table,\n",
    ")\n",
    "\n",
    "feature_table.visualizers.tabulate_seqs(data = rep_seqs.filtered_data).visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4846da6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'./results/denoising_stats.qza'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## save the output\n",
    "denoisetable.filtered_table.save('./results/denoisetable.qza')\n",
    "rep_seqs.filtered_data.save('./results/rep_seqs.qza')\n",
    "denoising_stats_all.save('./results/denoising_stats.qza')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
