{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "%reload_ext autoreload\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import packages\n",
    "import qiime2 as q2\n",
    "import os\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins import cutadapt\n",
    "from qiime2.plugins import dada2\n",
    "from qiime2.plugins import feature_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: WORKDIR=/home/nezapa/qiime-thesis\n"
     ]
    }
   ],
   "source": [
    "# define workdir\n",
    "%env WORKDIR /home/nezapa/qiime-thesis\n",
    "WORKDIR = os.environ.get(\"WORKDIR\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### IMPORT DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set manifest path\n",
    "manifest_path = './00.manifest.f.tsv'\n",
    "\n",
    "#import data\n",
    "single_end_demux = q2.Artifact.import_data('SampleData[SequencesWithQuality]', \n",
    "    view_type='SingleEndFastqManifestPhred33V2', \n",
    "    view=manifest_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-sbc7icnz')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 3cb8e5bc-f558-4169-a55a-6b4295eee010>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#summarise and visualise\n",
    "demux_summary = demux.visualizers.summarize(single_end_demux)\n",
    "demux_summary.visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>location</th>\n",
       "      <th>origin</th>\n",
       "      <th>year</th>\n",
       "      <th>parkelj</th>\n",
       "      <th>specimen</th>\n",
       "      <th>population</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P01A</th>\n",
       "      <td>P01A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa200_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa200</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P03A</th>\n",
       "      <td>P03A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa201_t0</td>\n",
       "      <td>2015</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa201</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P05A</th>\n",
       "      <td>P05A</td>\n",
       "      <td>Stobe</td>\n",
       "      <td>Paa204_t10d</td>\n",
       "      <td>2016</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa204</td>\n",
       "      <td>dolenjska linija</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P07A</th>\n",
       "      <td>P07A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa209_t0</td>\n",
       "      <td>2017</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa209</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P09A</th>\n",
       "      <td>P09A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa210_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa210</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P11A</th>\n",
       "      <td>P11A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa211_t0</td>\n",
       "      <td>2018</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa211</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P13A</th>\n",
       "      <td>P13A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa219_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa219</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P15A</th>\n",
       "      <td>P15A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa220_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa220</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P17A</th>\n",
       "      <td>P17A</td>\n",
       "      <td>Planinska_jama</td>\n",
       "      <td>Paa221_t0</td>\n",
       "      <td>2019</td>\n",
       "      <td>no</td>\n",
       "      <td>Paa221</td>\n",
       "      <td>linija reke Ljubljanice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P19A</th>\n",
       "      <td>P19A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB271_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB271</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P20A</th>\n",
       "      <td>P20A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PB272_parkelj</td>\n",
       "      <td>2016</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB272</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P21A</th>\n",
       "      <td>P21A</td>\n",
       "      <td>Jelsevnik</td>\n",
       "      <td>PB644_parkelj</td>\n",
       "      <td>2015</td>\n",
       "      <td>yes</td>\n",
       "      <td>PB644</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P22A</th>\n",
       "      <td>P22A</td>\n",
       "      <td>Obrsec</td>\n",
       "      <td>PAP20_parkelj</td>\n",
       "      <td>2020</td>\n",
       "      <td>yes</td>\n",
       "      <td>PAP20</td>\n",
       "      <td>훾rna podvrsta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample        location         origin  year parkelj specimen  \\\n",
       "sampleid                                                                \n",
       "P01A       P01A  Planinska_jama      Paa200_t0  2015      no   Paa200   \n",
       "P03A       P03A  Planinska_jama      Paa201_t0  2015      no   Paa201   \n",
       "P05A       P05A           Stobe    Paa204_t10d  2016      no   Paa204   \n",
       "P07A       P07A  Planinska_jama      Paa209_t0  2017      no   Paa209   \n",
       "P09A       P09A  Planinska_jama      Paa210_t0  2019      no   Paa210   \n",
       "P11A       P11A  Planinska_jama      Paa211_t0  2018      no   Paa211   \n",
       "P13A       P13A  Planinska_jama      Paa219_t0  2019      no   Paa219   \n",
       "P15A       P15A  Planinska_jama      Paa220_t0  2019      no   Paa220   \n",
       "P17A       P17A  Planinska_jama      Paa221_t0  2019      no   Paa221   \n",
       "P19A       P19A          Obrsec  PB271_parkelj  2016     yes    PB271   \n",
       "P20A       P20A          Obrsec  PB272_parkelj  2016     yes    PB272   \n",
       "P21A       P21A       Jelsevnik  PB644_parkelj  2015     yes    PB644   \n",
       "P22A       P22A          Obrsec  PAP20_parkelj  2020     yes    PAP20   \n",
       "\n",
       "                       population  \n",
       "sampleid                           \n",
       "P01A      linija reke Ljubljanice  \n",
       "P03A      linija reke Ljubljanice  \n",
       "P05A             dolenjska linija  \n",
       "P07A      linija reke Ljubljanice  \n",
       "P09A      linija reke Ljubljanice  \n",
       "P11A      linija reke Ljubljanice  \n",
       "P13A      linija reke Ljubljanice  \n",
       "P15A      linija reke Ljubljanice  \n",
       "P17A      linija reke Ljubljanice  \n",
       "P19A                훾rna podvrsta  \n",
       "P20A                훾rna podvrsta  \n",
       "P21A                훾rna podvrsta  \n",
       "P22A                훾rna podvrsta  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load and view metadata\n",
    "sample_metadata = q2.Metadata.load('../00.sample-metadata.tsv')\n",
    "sample_metadata.to_dataframe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### TRIM PRIMERS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P01A_0_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P01A_0_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P01A_0_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P01A_0_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.20 s (33 쨉s/read; 1.83 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  36,484\n",
      "Reads with adapters:                    36,141 (99.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   2 (0.0%)\n",
      "Reads written (passing filters):        36,482 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,867,659 bp\n",
      "Total written (filtered):     10,073,532 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 192 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 34.4%\n",
      "  C: 7.8%\n",
      "  G: 46.4%\n",
      "  T: 10.4%\n",
      "  none/other: 1.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t570.1\t0\t5\n",
      "4\t4\t142.5\t0\t3 1\n",
      "5\t12\t35.6\t1\t0 12\n",
      "6\t1\t8.9\t1\t0 1\n",
      "8\t1\t0.6\t1\t0 1\n",
      "20\t147\t0.0\t4\t142 5\n",
      "21\t1\t0.0\t4\t1\n",
      "22\t4\t0.0\t4\t0 1 3\n",
      "23\t1\t0.0\t4\t0 0 1\n",
      "25\t3\t0.0\t5\t0 0 1 0 1 1\n",
      "30\t1\t0.0\t6\t0 0 1\n",
      "37\t1\t0.0\t7\t0 0 0 0 0 1\n",
      "38\t1\t0.0\t7\t0 0 1\n",
      "44\t1\t0.0\t8\t0 0 0 0 0 0 1\n",
      "50\t1\t0.0\t10\t0 0 0 0 0 0 0 0 1\n",
      "65\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "67\t2\t0.0\t10\t0 0 0 0 1 0 0 1\n",
      "68\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "87\t1\t0.0\t10\t0 0 0 0 1\n",
      "91\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "299\t1\t0.0\t10\t0 0 0 0 0 0 0 0 1\n",
      "300\t1\t0.0\t10\t0 0 0 0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 35949 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t570.1\t0\t2\n",
      "4\t4\t142.5\t0\t2 2\n",
      "5\t5\t35.6\t1\t0 5\n",
      "10\t1\t0.0\t2\t0 0 1\n",
      "17\t2\t0.0\t3\t1 0 1\n",
      "18\t1\t0.0\t3\t0 0 1\n",
      "19\t8\t0.0\t3\t2 1 3 2\n",
      "20\t65\t0.0\t4\t19 23 21 0 2\n",
      "21\t1117\t0.0\t4\t489 606 19 1 2\n",
      "22\t34538\t0.0\t4\t32996 1388 88 56 10\n",
      "23\t195\t0.0\t4\t44 117 22 12\n",
      "24\t6\t0.0\t4\t0 2 4\n",
      "25\t1\t0.0\t5\t1\n",
      "26\t1\t0.0\t5\t0 0 1\n",
      "30\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "32\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "36\t1\t0.0\t7\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P03A_1_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P03A_1_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P03A_1_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P03A_1_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.64 s (24 쨉s/read; 2.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  67,647\n",
      "Reads with adapters:                    66,813 (98.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        67,647 (100.0%)\n",
      "\n",
      "Total basepairs processed:    20,283,542 bp\n",
      "Total written (filtered):     18,816,563 bp (92.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 43 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 76.7%\n",
      "  C: 11.6%\n",
      "  G: 7.0%\n",
      "  T: 4.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t1057.0\t0\t3\n",
      "4\t2\t264.2\t0\t1 1\n",
      "5\t14\t66.1\t1\t0 14\n",
      "19\t1\t0.0\t3\t1\n",
      "20\t12\t0.0\t4\t12\n",
      "21\t5\t0.0\t4\t2 3\n",
      "22\t1\t0.0\t4\t0 0 1\n",
      "23\t2\t0.0\t4\t0 1 0 1\n",
      "24\t1\t0.0\t4\t0 1\n",
      "25\t2\t0.0\t5\t0 1 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 66770 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t1057.0\t0\t6\n",
      "4\t3\t264.2\t0\t0 3\n",
      "5\t16\t66.1\t1\t1 15\n",
      "8\t1\t1.0\t1\t0 1\n",
      "10\t5\t0.1\t2\t0 0 5\n",
      "15\t1\t0.0\t3\t1\n",
      "16\t1\t0.0\t3\t0 1\n",
      "17\t2\t0.0\t3\t0 1 1\n",
      "18\t5\t0.0\t3\t2 1 2\n",
      "19\t23\t0.0\t3\t8 10 3 2\n",
      "20\t128\t0.0\t4\t42 56 30\n",
      "21\t2044\t0.0\t4\t964 1070 9 0 1\n",
      "22\t64191\t0.0\t4\t62794 1344 28 13 12\n",
      "23\t336\t0.0\t4\t74 204 44 13 1\n",
      "24\t3\t0.0\t4\t0 2 1\n",
      "26\t1\t0.0\t5\t1\n",
      "27\t2\t0.0\t5\t0 0 0 1 1\n",
      "28\t1\t0.0\t5\t0 0 0 0 1\n",
      "32\t1\t0.0\t6\t0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P05A_2_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P05A_2_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P05A_2_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P05A_2_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.27 s (26 쨉s/read; 2.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  48,673\n",
      "Reads with adapters:                    48,130 (98.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        48,673 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,573,760 bp\n",
      "Total written (filtered):     13,516,205 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 103 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 65.0%\n",
      "  C: 10.7%\n",
      "  G: 18.4%\n",
      "  T: 5.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t760.5\t0\t4\n",
      "4\t2\t190.1\t0\t2\n",
      "5\t3\t47.5\t1\t0 3\n",
      "19\t1\t0.0\t3\t0 1\n",
      "20\t72\t0.0\t4\t71 1\n",
      "21\t3\t0.0\t4\t1 2\n",
      "22\t1\t0.0\t4\t0 0 1\n",
      "23\t1\t0.0\t4\t0 0 1\n",
      "24\t3\t0.0\t4\t0 2 0 1\n",
      "40\t1\t0.0\t8\t0 0 0 0 0 0 0 1\n",
      "53\t2\t0.0\t10\t0 0 2\n",
      "57\t4\t0.0\t10\t0 0 2 0 1 1\n",
      "59\t1\t0.0\t10\t0 0 0 1\n",
      "66\t1\t0.0\t10\t0 0 0 0 1\n",
      "96\t1\t0.0\t10\t0 0 0 1\n",
      "158\t1\t0.0\t10\t0 0 1\n",
      "234\t1\t0.0\t10\t0 0 0 0 1\n",
      "266\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 48027 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t760.5\t0\t7\n",
      "4\t11\t190.1\t0\t3 8\n",
      "5\t19\t47.5\t1\t0 19\n",
      "10\t1\t0.0\t2\t0 1\n",
      "11\t1\t0.0\t2\t1\n",
      "18\t5\t0.0\t3\t1 1 1 1 1\n",
      "19\t10\t0.0\t3\t1 4 4 1\n",
      "20\t66\t0.0\t4\t30 19 16 1\n",
      "21\t1375\t0.0\t4\t663 699 12 1\n",
      "22\t46271\t0.0\t4\t45033 1176 37 14 11\n",
      "23\t257\t0.0\t4\t49 162 32 14\n",
      "24\t1\t0.0\t4\t0 0 1\n",
      "25\t3\t0.0\t5\t2 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P07A_3_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P07A_3_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P07A_3_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P07A_3_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.21 s (32 쨉s/read; 1.85 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  37,257\n",
      "Reads with adapters:                    37,016 (99.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        37,257 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,152,827 bp\n",
      "Total written (filtered):     10,339,596 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 45 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 44.4%\n",
      "  C: 15.6%\n",
      "  G: 22.2%\n",
      "  T: 17.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t582.1\t0\t3\n",
      "4\t1\t145.5\t0\t0 1\n",
      "5\t7\t36.4\t1\t0 7\n",
      "20\t18\t0.0\t4\t18\n",
      "21\t1\t0.0\t4\t0 0 1\n",
      "22\t5\t0.0\t4\t0 3 1 1\n",
      "23\t3\t0.0\t4\t0 1 2\n",
      "25\t1\t0.0\t5\t0 1\n",
      "31\t1\t0.0\t6\t0 0 1\n",
      "43\t1\t0.0\t8\t0 0 0 0 0 0 0 0 1\n",
      "49\t1\t0.0\t9\t0 0 0 0 0 0 1\n",
      "60\t1\t0.0\t10\t0 0 0 0 1\n",
      "106\t1\t0.0\t10\t0 0 0 0 0 0 0 1\n",
      "114\t1\t0.0\t10\t0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 36971 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t582.1\t0\t2\n",
      "4\t1\t145.5\t0\t0 1\n",
      "5\t5\t36.4\t1\t0 5\n",
      "10\t2\t0.0\t2\t0 0 2\n",
      "16\t2\t0.0\t3\t1 1\n",
      "17\t1\t0.0\t3\t1\n",
      "18\t2\t0.0\t3\t1 1\n",
      "19\t2\t0.0\t3\t1 1\n",
      "20\t64\t0.0\t4\t18 24 22\n",
      "21\t1088\t0.0\t4\t490 586 12\n",
      "22\t35592\t0.0\t4\t34555 987 32 8 10\n",
      "23\t204\t0.0\t4\t47 122 27 8\n",
      "24\t3\t0.0\t4\t1 1 1\n",
      "27\t1\t0.0\t5\t0 0 1\n",
      "29\t1\t0.0\t5\t1\n",
      "45\t1\t0.0\t9\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P09A_4_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P09A_4_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P09A_4_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P09A_4_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.03 s (27 쨉s/read; 2.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  37,537\n",
      "Reads with adapters:                    37,312 (99.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        37,537 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,246,904 bp\n",
      "Total written (filtered):     10,427,401 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 44 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 50.0%\n",
      "  C: 9.1%\n",
      "  G: 27.3%\n",
      "  T: 13.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t586.5\t0\t3\n",
      "4\t1\t146.6\t0\t1\n",
      "5\t2\t36.7\t1\t0 2\n",
      "13\t1\t0.0\t2\t0 0 0 1\n",
      "19\t1\t0.0\t3\t1\n",
      "20\t28\t0.0\t4\t25 3\n",
      "21\t3\t0.0\t4\t1 2\n",
      "22\t2\t0.0\t4\t0 1 0 1\n",
      "23\t2\t0.0\t4\t0 0 0 2\n",
      "68\t1\t0.0\t10\t0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 37268 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "4\t2\t146.6\t0\t0 2\n",
      "5\t6\t36.7\t1\t1 5\n",
      "10\t1\t0.0\t2\t0 0 1\n",
      "17\t1\t0.0\t3\t1\n",
      "18\t2\t0.0\t3\t0 1 0 0 1\n",
      "19\t9\t0.0\t3\t3 3 3\n",
      "20\t65\t0.0\t4\t23 24 17 1\n",
      "21\t1127\t0.0\t4\t514 592 18 1 2\n",
      "22\t35837\t0.0\t4\t34891 914 19 5 8\n",
      "23\t218\t0.0\t4\t45 144 24 5\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P11A_5_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P11A_5_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P11A_5_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P11A_5_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.16 s (24 쨉s/read; 2.46 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  47,434\n",
      "Reads with adapters:                    46,733 (98.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        47,434 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,220,009 bp\n",
      "Total written (filtered):     13,194,010 bp (92.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 28 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 39.3%\n",
      "  C: 14.3%\n",
      "  G: 32.1%\n",
      "  T: 14.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t741.2\t0\t6\n",
      "5\t4\t46.3\t1\t0 4\n",
      "6\t1\t11.6\t1\t0 1\n",
      "19\t1\t0.0\t3\t1\n",
      "20\t6\t0.0\t4\t6\n",
      "21\t3\t0.0\t4\t1 1 1\n",
      "23\t3\t0.0\t4\t0 1 2\n",
      "24\t2\t0.0\t4\t0 1 1\n",
      "25\t1\t0.0\t5\t0 0 0 0 1\n",
      "42\t1\t0.0\t8\t0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 46705 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t741.2\t0\t5\n",
      "4\t11\t185.3\t0\t2 9\n",
      "5\t18\t46.3\t1\t1 17\n",
      "10\t4\t0.0\t2\t0 0 4\n",
      "11\t1\t0.0\t2\t1\n",
      "16\t1\t0.0\t3\t1\n",
      "17\t2\t0.0\t3\t1 0 1\n",
      "18\t1\t0.0\t3\t1\n",
      "19\t9\t0.0\t3\t5 0 3 1\n",
      "20\t86\t0.0\t4\t33 31 22\n",
      "21\t1368\t0.0\t4\t636 721 9 0 2\n",
      "22\t44921\t0.0\t4\t43929 950 7 26 9\n",
      "23\t272\t0.0\t4\t63 177 26 6\n",
      "24\t3\t0.0\t4\t0 1 1 0 1\n",
      "26\t1\t0.0\t5\t0 0 1\n",
      "27\t1\t0.0\t5\t0 0 0 1\n",
      "39\t1\t0.0\t7\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P13A_6_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P13A_6_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P13A_6_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P13A_6_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.12 s (37 쨉s/read; 1.64 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  30,587\n",
      "Reads with adapters:                    29,962 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        30,587 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,131,370 bp\n",
      "Total written (filtered):      8,473,611 bp (92.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 88 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.9%\n",
      "  C: 30.7%\n",
      "  G: 45.5%\n",
      "  T: 8.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t477.9\t0\t4\n",
      "4\t1\t119.5\t0\t1\n",
      "5\t4\t29.9\t1\t1 3\n",
      "6\t3\t7.5\t1\t0 3\n",
      "19\t1\t0.0\t3\t1\n",
      "20\t62\t0.0\t4\t47 15\n",
      "21\t3\t0.0\t4\t1 2\n",
      "22\t1\t0.0\t4\t0 0 1\n",
      "23\t3\t0.0\t4\t0 0 2 1\n",
      "37\t1\t0.0\t7\t0 0 0 0 1\n",
      "63\t3\t0.0\t10\t0 0 0 1 1 0 0 0 1\n",
      "68\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "155\t1\t0.0\t10\t0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 29874 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t477.9\t0\t7\n",
      "4\t5\t119.5\t0\t0 5\n",
      "5\t18\t29.9\t1\t2 16\n",
      "17\t1\t0.0\t3\t1\n",
      "18\t2\t0.0\t3\t1 1\n",
      "19\t13\t0.0\t3\t5 2 3 3\n",
      "20\t47\t0.0\t4\t17 16 14\n",
      "21\t887\t0.0\t4\t408 450 24 3 2\n",
      "22\t28727\t0.0\t4\t26695 1966 52 5 9\n",
      "23\t159\t0.0\t4\t28 100 25 6\n",
      "24\t7\t0.0\t4\t1 1 1 4\n",
      "26\t1\t0.0\t5\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P15A_7_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P15A_7_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P15A_7_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P15A_7_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.05 s (33 쨉s/read; 1.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  31,754\n",
      "Reads with adapters:                    31,519 (99.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        31,754 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,511,630 bp\n",
      "Total written (filtered):      8,819,516 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 59 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 49.2%\n",
      "  C: 1.7%\n",
      "  G: 45.8%\n",
      "  T: 3.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t496.2\t0\t3\n",
      "4\t1\t124.0\t0\t0 1\n",
      "5\t6\t31.0\t1\t0 6\n",
      "19\t1\t0.0\t3\t1\n",
      "20\t33\t0.0\t4\t32 1\n",
      "21\t5\t0.0\t4\t3 2\n",
      "22\t2\t0.0\t4\t0 0 1 1\n",
      "23\t1\t0.0\t4\t0 0 1\n",
      "24\t1\t0.0\t4\t0 0 1\n",
      "25\t4\t0.0\t5\t0 0 0 0 1 3\n",
      "26\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "38\t1\t0.0\t7\t0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 31460 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t496.2\t0\t2\n",
      "4\t2\t124.0\t0\t2\n",
      "5\t6\t31.0\t1\t0 6\n",
      "12\t1\t0.0\t2\t1\n",
      "17\t3\t0.0\t3\t3\n",
      "18\t1\t0.0\t3\t1\n",
      "19\t4\t0.0\t3\t0 1 3\n",
      "20\t55\t0.0\t4\t19 19 16 1\n",
      "21\t969\t0.0\t4\t453 500 12 1 3\n",
      "22\t30244\t0.0\t4\t29486 717 27 9 5\n",
      "23\t167\t0.0\t4\t36 100 23 8\n",
      "24\t2\t0.0\t4\t0 2\n",
      "28\t3\t0.0\t5\t1 0 1 1\n",
      "41\t1\t0.0\t8\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P17A_8_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P17A_8_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P17A_8_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P17A_8_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.23 s (28 쨉s/read; 2.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  44,620\n",
      "Reads with adapters:                    44,394 (99.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        44,620 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,304,016 bp\n",
      "Total written (filtered):     12,329,159 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 414 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 68.1%\n",
      "  C: 4.1%\n",
      "  G: 9.9%\n",
      "  T: 17.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t697.2\t0\t2\n",
      "4\t2\t174.3\t0\t2\n",
      "5\t7\t43.6\t1\t1 6\n",
      "6\t1\t10.9\t1\t0 1\n",
      "18\t1\t0.0\t3\t0 0 0 1\n",
      "19\t2\t0.0\t3\t1 1\n",
      "20\t356\t0.0\t4\t349 7\n",
      "21\t3\t0.0\t4\t1 1 1\n",
      "22\t2\t0.0\t4\t0 2\n",
      "23\t3\t0.0\t4\t0 1 2\n",
      "24\t8\t0.0\t4\t0 0 1 5 2\n",
      "25\t13\t0.0\t5\t0 0 1 8 3 1\n",
      "26\t2\t0.0\t5\t0 0 0 0 1 1\n",
      "30\t1\t0.0\t6\t0 0 0 1\n",
      "31\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "34\t1\t0.0\t6\t0 0 1\n",
      "45\t1\t0.0\t9\t0 0 0 0 0 0 0 1\n",
      "48\t1\t0.0\t9\t0 0 0 0 0 0 0 0 0 1\n",
      "68\t2\t0.0\t10\t0 0 0 0 1 1\n",
      "69\t2\t0.0\t10\t0 0 0 0 0 1 1\n",
      "72\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "75\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "119\t1\t0.0\t10\t0 0 0 0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 43980 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t697.2\t0\t1\n",
      "4\t1\t174.3\t0\t1\n",
      "5\t6\t43.6\t1\t1 5\n",
      "13\t1\t0.0\t2\t1\n",
      "16\t1\t0.0\t3\t0 1\n",
      "17\t1\t0.0\t3\t0 0 1\n",
      "18\t3\t0.0\t3\t1 0 1 1\n",
      "19\t6\t0.0\t3\t2 2 1 1\n",
      "20\t91\t0.0\t4\t33 22 35 1\n",
      "21\t1363\t0.0\t4\t589 725 45 3 1\n",
      "22\t42208\t0.0\t4\t40088 1974 118 14 14\n",
      "23\t289\t0.0\t4\t53 176 42 16 2\n",
      "24\t3\t0.0\t4\t0 0 2 0 1\n",
      "25\t2\t0.0\t5\t1 0 1\n",
      "26\t1\t0.0\t5\t0 0 1\n",
      "27\t1\t0.0\t5\t0 0 0 1\n",
      "29\t1\t0.0\t5\t0 1\n",
      "32\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P19A_9_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P19A_9_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P19A_9_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P19A_9_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.06 s (26 쨉s/read; 2.30 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  40,719\n",
      "Reads with adapters:                    39,948 (98.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        40,719 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,202,958 bp\n",
      "Total written (filtered):     11,325,923 bp (92.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 58 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 70.7%\n",
      "  C: 3.4%\n",
      "  G: 8.6%\n",
      "  T: 17.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t636.2\t0\t3\n",
      "4\t1\t159.1\t0\t1\n",
      "5\t3\t39.8\t1\t1 2\n",
      "9\t1\t0.2\t1\t0 1\n",
      "16\t1\t0.0\t3\t0 0 0 1\n",
      "19\t1\t0.0\t3\t1\n",
      "20\t22\t0.0\t4\t22\n",
      "21\t5\t0.0\t4\t4 0 1\n",
      "22\t2\t0.0\t4\t0 2\n",
      "23\t1\t0.0\t4\t0 0 1\n",
      "24\t5\t0.0\t4\t0 0 0 2 3\n",
      "25\t8\t0.0\t5\t0 0 0 2 4 2\n",
      "26\t3\t0.0\t5\t0 0 0 0 3\n",
      "67\t1\t0.0\t10\t0 0 0 0 0 0 0 1\n",
      "68\t1\t0.0\t10\t0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 39890 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t636.2\t0\t5\n",
      "4\t11\t159.1\t0\t0 11\n",
      "5\t20\t39.8\t1\t1 19\n",
      "6\t1\t9.9\t1\t0 1\n",
      "10\t3\t0.0\t2\t0 0 3\n",
      "13\t1\t0.0\t2\t1\n",
      "17\t2\t0.0\t3\t2\n",
      "18\t1\t0.0\t3\t0 1\n",
      "19\t10\t0.0\t3\t4 4 1 1\n",
      "20\t58\t0.0\t4\t15 20 23\n",
      "21\t1168\t0.0\t4\t550 608 9 0 1\n",
      "22\t38380\t0.0\t4\t37482 844 33 11 10\n",
      "23\t226\t0.0\t4\t49 151 20 5 1\n",
      "26\t1\t0.0\t5\t0 0 0 0 1\n",
      "29\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "32\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "35\t1\t0.0\t7\t0 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P20A_10_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P20A_10_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P20A_10_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P20A_10_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.37 s (29 쨉s/read; 2.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  46,539\n",
      "Reads with adapters:                    46,133 (99.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        46,539 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,926,053 bp\n",
      "Total written (filtered):     12,912,755 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 66 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 27.3%\n",
      "  G: 30.3%\n",
      "  T: 15.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t727.2\t0\t3\n",
      "4\t5\t181.8\t0\t5\n",
      "5\t6\t45.4\t1\t2 4\n",
      "20\t44\t0.0\t4\t28 15 0 0 1\n",
      "21\t3\t0.0\t4\t0 2 1\n",
      "23\t1\t0.0\t4\t0 1\n",
      "27\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "106\t1\t0.0\t10\t0 0 0 1\n",
      "117\t1\t0.0\t10\t0 0 0 0 0 0 0 0 0 1\n",
      "142\t1\t0.0\t10\t0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 46067 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t727.2\t0\t4\n",
      "4\t4\t181.8\t0\t0 4\n",
      "5\t6\t45.4\t1\t1 5\n",
      "6\t1\t11.4\t1\t0 1\n",
      "10\t2\t0.0\t2\t0 0 2\n",
      "11\t1\t0.0\t2\t0 0 1\n",
      "17\t3\t0.0\t3\t3\n",
      "18\t2\t0.0\t3\t0 1 1\n",
      "19\t10\t0.0\t3\t4 1 4 1\n",
      "20\t77\t0.0\t4\t21 31 24 0 1\n",
      "21\t1374\t0.0\t4\t607 735 26 6\n",
      "22\t44311\t0.0\t4\t42781 1400 100 17 13\n",
      "23\t268\t0.0\t4\t56 172 33 6 1\n",
      "24\t1\t0.0\t4\t0 1\n",
      "27\t1\t0.0\t5\t0 1\n",
      "29\t2\t0.0\t5\t1 1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P21A_11_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P21A_11_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P21A_11_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P21A_11_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.17 s (24 쨉s/read; 2.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  49,197\n",
      "Reads with adapters:                    48,542 (98.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        49,197 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,740,883 bp\n",
      "Total written (filtered):     13,674,891 bp (92.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 75 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 41.3%\n",
      "  C: 33.3%\n",
      "  G: 22.7%\n",
      "  T: 2.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t768.7\t0\t3\n",
      "4\t23\t192.2\t0\t1 22\n",
      "5\t1\t48.0\t1\t1\n",
      "6\t1\t12.0\t1\t0 1\n",
      "18\t1\t0.0\t3\t0 0 0 0 1\n",
      "20\t33\t0.0\t4\t33\n",
      "21\t3\t0.0\t4\t3\n",
      "22\t2\t0.0\t4\t0 0 2\n",
      "23\t1\t0.0\t4\t0 1\n",
      "24\t2\t0.0\t4\t0 0 1 1\n",
      "47\t1\t0.0\t9\t0 0 0 1\n",
      "52\t1\t0.0\t10\t0 0 0 0 0 0 0 1\n",
      "67\t1\t0.0\t10\t0 0 0 0 0 1\n",
      "272\t1\t0.0\t10\t0 0 1\n",
      "298\t1\t0.0\t10\t0 0 0 0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 48467 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t768.7\t0\t5\n",
      "4\t13\t192.2\t0\t3 10\n",
      "5\t12\t48.0\t1\t2 10\n",
      "6\t1\t12.0\t1\t0 1\n",
      "8\t1\t0.8\t1\t0 1\n",
      "10\t2\t0.0\t2\t0 0 2\n",
      "13\t1\t0.0\t2\t1\n",
      "15\t1\t0.0\t3\t0 1\n",
      "17\t2\t0.0\t3\t1 0 1\n",
      "18\t5\t0.0\t3\t1 1 1 0 2\n",
      "19\t12\t0.0\t3\t4 4 1 3\n",
      "20\t72\t0.0\t4\t22 26 24\n",
      "21\t1504\t0.0\t4\t672 803 22 3 4\n",
      "22\t46556\t0.0\t4\t45202 1270 56 16 12\n",
      "23\t273\t0.0\t4\t57 169 31 15 1\n",
      "24\t1\t0.0\t4\t0 0 1\n",
      "26\t1\t0.0\t5\t0 0 1\n",
      "30\t2\t0.0\t6\t0 0 0 0 0 2\n",
      "32\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "33\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "37\t1\t0.0\t7\t1\n",
      "\n",
      "Command: cutadapt --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P22A_12_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P22A_12_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.0 with Python 3.8.13\n",
      "Command line parameters: --cores 28 --error-rate 0.2 --times 1 --overlap 3 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-634b3onz/P22A_12_L001_R1_001.fastq.gz --adapter GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC --front TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG /tmp/qiime2-archive-ntstfe3z/8b823ce2-f03c-4ca9-a3d5-617a5131fc28/data/P22A_12_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 28 cores ...\n",
      "Finished in 1.56 s (23 쨉s/read; 2.63 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  68,413\n",
      "Reads with adapters:                    68,164 (99.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        68,413 (100.0%)\n",
      "\n",
      "Total basepairs processed:    20,457,195 bp\n",
      "Total written (filtered):     18,960,848 bp (92.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC; Type: regular 3'; Length: 54; Trimmed: 406 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9; 50-54 bp: 10\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 89.4%\n",
      "  C: 3.4%\n",
      "  G: 5.7%\n",
      "  T: 1.5%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'A' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t1069.0\t0\t7\n",
      "4\t3\t267.2\t0\t3\n",
      "5\t10\t66.8\t1\t0 10\n",
      "7\t1\t4.2\t1\t0 1\n",
      "19\t1\t0.0\t3\t0 1\n",
      "20\t368\t0.0\t4\t368\n",
      "21\t6\t0.0\t4\t3 3\n",
      "22\t1\t0.0\t4\t0 1\n",
      "23\t5\t0.0\t4\t0 0 5\n",
      "24\t1\t0.0\t4\t0 0 0 1\n",
      "25\t2\t0.0\t5\t0 0 1 0 0 1\n",
      "31\t1\t0.0\t6\t0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG; Type: regular 5'; Length: 55; Trimmed: 67758 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-29 bp: 5; 30-34 bp: 6; 35-39 bp: 7; 40-44 bp: 8; 45-49 bp: 9\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t1069.0\t0\t2\n",
      "4\t4\t267.2\t0\t0 4\n",
      "5\t5\t66.8\t1\t0 5\n",
      "13\t1\t0.0\t2\t1\n",
      "16\t1\t0.0\t3\t0 1\n",
      "17\t1\t0.0\t3\t1\n",
      "18\t5\t0.0\t3\t2 2 1\n",
      "19\t16\t0.0\t3\t5 2 5 4\n",
      "20\t119\t0.0\t4\t40 37 41 1\n",
      "21\t2050\t0.0\t4\t948 1076 25 1\n",
      "22\t65154\t0.0\t4\t63586 1495 40 18 15\n",
      "23\t398\t0.0\t4\t101 247 35 11 4\n",
      "25\t1\t0.0\t5\t0 0 1\n",
      "26\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n"
     ]
    }
   ],
   "source": [
    "# adaoter has to be a reverse complement\n",
    "single_end_trimmed = cutadapt.methods.trim_single(\n",
    "    demultiplexed_sequences = single_end_demux,\n",
    "    front = ['TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGNNNNNCCTACGGGNGGCWGCAG'],\n",
    "    adapter = ['GGATTAGATACCCBDGTAGTCTGTCTCTTATACACATCTCCGAGCCCACGAGAC'], \n",
    "    error_rate = 0.2,\n",
    "    cores = 28\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### START DADA2 MERGING AND CLUSTERING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada_single.R /tmp/qiime2-archive-pc9g_2d4/cbfda859-8fa5-4e3e-a357-268c6c981652/data /tmp/tmphm07yhdj/output.tsv.biom /tmp/tmphm07yhdj/track.tsv /tmp/tmphm07yhdj 260 0 2.0 2 Inf pseudo consensus 1.0 14 1000000 NULL 16\n",
      "\n",
      "R version 4.1.3 (2022-03-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rcpp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DADA2: 1.22.0 / Rcpp: 1.0.8.3 / RcppParallel: 5.1.5 \n",
      "1) Filtering .............\n",
      "2) Learning Error Rates\n",
      "143454740 total bases in 551749 reads from 13 samples will be used for learning the error rates.\n",
      "3) Denoise samples .............\n",
      "  Pseudo-pool step .............\n",
      "4) Remove chimeras (method = consensus)\n",
      "5) Report read numbers through the pipeline\n",
      "6) Write output\n"
     ]
    }
   ],
   "source": [
    "# set trunc-len based on the Interactive Quality Plot in datasummary.qzv\n",
    "# decide where the sequence quality drops below an acceptable level\n",
    "# this step can take a few minutes\n",
    "denoisetable_all, rep_seqs_all, denoising_stats_all = dada2.methods.denoise_single(\n",
    "  demultiplexed_seqs = single_end_trimmed.trimmed_sequences,\n",
    "  trunc_len = 260,\n",
    "  pooling_method = 'pseudo',\n",
    "  n_threads = 14\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### FILTERING BY ABUNDANCE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set the min_frequency\n",
    "denoisetable= feature_table.methods.filter_features(\n",
    "    table = denoisetable_all,\n",
    "    min_frequency = 20,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-rxqo2oxf')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: dde7863a-51be-44eb-a6a9-e07d326499d7>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# summarise and visualise\n",
    "feature_table.visualizers.summarize(\n",
    "    table = denoisetable.filtered_table,\n",
    "    sample_metadata = sample_metadata\n",
    ").visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-nxqr52q2')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: ea963cb7-ecb4-4ec7-8231-f1b0af76f4f9>"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# visualise the new set of representative sequences\n",
    "rep_seqs = feature_table.methods.filter_seqs(\n",
    "    data = rep_seqs_all,\n",
    "    table = denoisetable.filtered_table,\n",
    ")\n",
    "\n",
    "feature_table.visualizers.tabulate_seqs(data = rep_seqs.filtered_data).visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'./results/denoising_stats.qza'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## save the output\n",
    "denoisetable.filtered_table.save('./results/denoisetable.qza')\n",
    "rep_seqs.filtered_data.save('./results/rep_seqs.qza')\n",
    "denoising_stats_all.save('./results/denoising_stats.qza')"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "9cfc1ea77cca5050ec46e353aa698c92876315d524268f42327e9c97aacfbb5a"
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
